{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "f52de6cd-520a-4ee6-bace-70552ddca45f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import pprint\n",
    "sns.set(style=\"white\", context='poster', font_scale=3)\n",
    "import math\n",
    "from scipy.integrate import solve_ivp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "7998d39b-e5fd-4289-bbf7-d1e547f378a1",
   "metadata": {},
   "outputs": [],
   "source": [
    "%config InlineBackend.figure_formats = ['svg']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "28fc9eec-fe5f-498b-a32b-83330606b2f5",
   "metadata": {},
   "outputs": [],
   "source": [
    "import seaborn_figure_settings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "002a40bc-57d9-44fb-9784-754552254804",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "save_dpath = os.path.join('../notebooks/figures/', 'Fig1')\n",
    "os.makedirs(save_dpath,exist_ok=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "699a8a28-8879-48c5-ae69-7ae66dbe4576",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<svg  width=\"220\" height=\"55\"><rect x=\"0\" y=\"0\" width=\"55\" height=\"55\" style=\"fill:#882255;stroke-width:2;stroke:rgb(255,255,255)\"/><rect x=\"55\" y=\"0\" width=\"55\" height=\"55\" style=\"fill:#cc6677;stroke-width:2;stroke:rgb(255,255,255)\"/><rect x=\"110\" y=\"0\" width=\"55\" height=\"55\" style=\"fill:#332288;stroke-width:2;stroke:rgb(255,255,255)\"/><rect x=\"165\" y=\"0\" width=\"55\" height=\"55\" style=\"fill:#44aa99;stroke-width:2;stroke:rgb(255,255,255)\"/></svg>"
      ],
      "text/plain": [
       "[(0.5333333333333333, 0.13333333333333333, 0.3333333333333333),\n",
       " (0.8, 0.4, 0.4666666666666667),\n",
       " (0.2, 0.13333333333333333, 0.5333333333333333),\n",
       " (0.26666666666666666, 0.6666666666666666, 0.6)]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gorder = [ 'Strong', 'Sustained', 'Inhibited', 'Weak', ]\n",
    "gpalette = ['#882255', '#CC6677', '#332288', \n",
    "             '#44AA99',  ]\n",
    "\n",
    "sns.color_palette(gpalette)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "63aa65ef-b3ee-4bb6-a027-2bb6bfba610a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['model_store_analyze_monte_biogeo_multi.ipynb',\n",
       " 'model_store_analyze_monte_het_analyze_flux.ipynb',\n",
       " 'model_store_analyze_monte_biogeo_create_files.ipynb',\n",
       " 'model_store_analyze_monte_ponly_curves.ipynb',\n",
       " 'Overflow_stats.xlsx',\n",
       " '10cc_fit_per_model.ipynb',\n",
       " 'model_store_analyze_monte_curves_cleanup.ipynb',\n",
       " 'figures',\n",
       " 'model_store_analyze_monte_curves.ipynb',\n",
       " 'sample_df.csv.gz',\n",
       " 'model_store_analyze_monte_het_params.ipynb',\n",
       " '.ipynb_checkpoints',\n",
       " 'model_store_analyze_monte_PCA.ipynb',\n",
       " 'percent_analysis.xlsx',\n",
       " 'phylodata',\n",
       " '__pycache__',\n",
       " 'verastile_vpros_multi.ipynb',\n",
       " 'top_runids.csv',\n",
       " 'model_store_analyze_monte_het_ROS.ipynb',\n",
       " 'qc_describe.csv',\n",
       " 'model_store_analyze_monte_het_analyze_flux_all.ipynb',\n",
       " 'figures_dynamics_old',\n",
       " 'verastile_vpros.ipynb',\n",
       " 'seaborn_figure_settings.py',\n",
       " 'CC10_phylogenetic_tree.ipynb',\n",
       " 'model_store_analyze_monte_ponly_params.ipynb',\n",
       " 'model_store_analyze_monte_curves_top.ipynb',\n",
       " 'model_store_analyze_monte_het_OVERFLOW.ipynb',\n",
       " 'create_venn.ipynb',\n",
       " '10cc_data_for_pca_biogeo.csv.gz',\n",
       " 'model_store_analyze_monte_het_EXOENZYME.ipynb',\n",
       " 'model_store_analyze_monte_biogeo.ipynb',\n",
       " 'model_store_analyze_monte_biogeo_full.ipynb',\n",
       " 'model_store_het_params_corr.ipynb']"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "os.listdir()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "c5f500a2-0eb0-40df-bc3f-13d377422d76",
   "metadata": {},
   "outputs": [],
   "source": [
    "pairwise_dist_df = pd.read_excel( 'phylodata/phyloDist.xlsx', sheet_name='phyloDist')\n",
    "metadata_df = pd.read_excel( 'phylodata/phyloDist.xlsx', sheet_name='metadata')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "9e1500e2-1ad1-4fdc-972b-d58239c3c91d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Group</th>\n",
       "      <th>Name</th>\n",
       "      <th>NCBI taxonomy id</th>\n",
       "      <th>strain_nickname</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Strong</td>\n",
       "      <td>Alteromonas macleodii</td>\n",
       "      <td>28108</td>\n",
       "      <td>HOT1A3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Strong</td>\n",
       "      <td>Pseudoalteromonas haloplanktis</td>\n",
       "      <td>326442</td>\n",
       "      <td>CIP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Sustained</td>\n",
       "      <td>Ruegeria pomeroyi</td>\n",
       "      <td>246200</td>\n",
       "      <td>DSS-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Sustained</td>\n",
       "      <td>Sulfitobacter pseudonitzschiae</td>\n",
       "      <td>1402135</td>\n",
       "      <td>nitzsch</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Weak</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>83333</td>\n",
       "      <td>25922</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Weak</td>\n",
       "      <td>Marinovum HOT5_F3</td>\n",
       "      <td>367771</td>\n",
       "      <td>HOT5_F3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Inhibited</td>\n",
       "      <td>Marinobacter adhaerens</td>\n",
       "      <td>225937</td>\n",
       "      <td>HP15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Weak</td>\n",
       "      <td>Roseovarius HOT5_C3</td>\n",
       "      <td>215743</td>\n",
       "      <td>HOT5_C3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Inhibited</td>\n",
       "      <td>Phaeobacter gallaeciensis</td>\n",
       "      <td>1423144</td>\n",
       "      <td>Pgall</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Group                            Name  NCBI taxonomy id strain_nickname\n",
       "0     Strong           Alteromonas macleodii             28108          HOT1A3\n",
       "1     Strong  Pseudoalteromonas haloplanktis            326442             CIP\n",
       "2  Sustained               Ruegeria pomeroyi            246200           DSS-3\n",
       "3  Sustained  Sulfitobacter pseudonitzschiae           1402135         nitzsch\n",
       "4       Weak                Escherichia coli             83333           25922\n",
       "5       Weak               Marinovum HOT5_F3            367771         HOT5_F3\n",
       "6  Inhibited          Marinobacter adhaerens            225937            HP15\n",
       "7       Weak             Roseovarius HOT5_C3            215743         HOT5_C3\n",
       "8  Inhibited       Phaeobacter gallaeciensis           1423144           Pgall"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "af9eb3f7-5d8e-4ac8-8ce2-de878e8dcede",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Group</th>\n",
       "      <th>Name</th>\n",
       "      <th>NCBI taxonomy id</th>\n",
       "      <th>strain_nickname</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Strong</td>\n",
       "      <td>Alteromonas macleodii</td>\n",
       "      <td>28108</td>\n",
       "      <td>HOT1A3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Strong</td>\n",
       "      <td>Pseudoalteromonas haloplanktis</td>\n",
       "      <td>326442</td>\n",
       "      <td>CIP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Sustained</td>\n",
       "      <td>Ruegeria pomeroyi</td>\n",
       "      <td>246200</td>\n",
       "      <td>DSS-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Sustained</td>\n",
       "      <td>Sulfitobacter pseudonitzschiae</td>\n",
       "      <td>1402135</td>\n",
       "      <td>nitzsch</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Weak</td>\n",
       "      <td>Marinovum HOT5_F3</td>\n",
       "      <td>367771</td>\n",
       "      <td>HOT5_F3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Inhibited</td>\n",
       "      <td>Marinobacter adhaerens</td>\n",
       "      <td>225937</td>\n",
       "      <td>HP15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Weak</td>\n",
       "      <td>Roseovarius HOT5_C3</td>\n",
       "      <td>215743</td>\n",
       "      <td>HOT5_C3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Inhibited</td>\n",
       "      <td>Phaeobacter gallaeciensis</td>\n",
       "      <td>1423144</td>\n",
       "      <td>Pgall</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Group                            Name  NCBI taxonomy id strain_nickname\n",
       "0     Strong           Alteromonas macleodii             28108          HOT1A3\n",
       "1     Strong  Pseudoalteromonas haloplanktis            326442             CIP\n",
       "2  Sustained               Ruegeria pomeroyi            246200           DSS-3\n",
       "3  Sustained  Sulfitobacter pseudonitzschiae           1402135         nitzsch\n",
       "5       Weak               Marinovum HOT5_F3            367771         HOT5_F3\n",
       "6  Inhibited          Marinobacter adhaerens            225937            HP15\n",
       "7       Weak             Roseovarius HOT5_C3            215743         HOT5_C3\n",
       "8  Inhibited       Phaeobacter gallaeciensis           1423144           Pgall"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# remove e coli\n",
    "metadata_df = metadata_df.loc[~metadata_df['Name'].isin(['Escherichia coli'])]\n",
    "metadata_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0d00888d-1480-4d5e-8b4a-278eb6148042",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "81b47d7b-fbaf-4973-8751-a2f96749b4d6",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_matrix_df = pd.read_excel( 'phylodata/genes.xlsx')\n",
    "trait_matrix_df = pd.read_excel( 'phylodata/traits.xlsx')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "0e082909-5e07-4007-bf5c-8ce39ed60d51",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>nitzsch</th>\n",
       "      <th>Pgall</th>\n",
       "      <th>DSS-3</th>\n",
       "      <th>HOT5_F3</th>\n",
       "      <th>HOT5_C3</th>\n",
       "      <th>HOT1A3</th>\n",
       "      <th>CIP</th>\n",
       "      <th>25922</th>\n",
       "      <th>HP15</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>nitzsch</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.157894</td>\n",
       "      <td>0.164756</td>\n",
       "      <td>0.171875</td>\n",
       "      <td>0.202754</td>\n",
       "      <td>2.537937</td>\n",
       "      <td>2.552027</td>\n",
       "      <td>2.492261</td>\n",
       "      <td>2.456501</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pgall</th>\n",
       "      <td>0.157894</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.099989</td>\n",
       "      <td>0.107108</td>\n",
       "      <td>0.137987</td>\n",
       "      <td>2.473170</td>\n",
       "      <td>2.487260</td>\n",
       "      <td>2.427494</td>\n",
       "      <td>2.391734</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DSS-3</th>\n",
       "      <td>0.164756</td>\n",
       "      <td>0.099989</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.106181</td>\n",
       "      <td>0.137060</td>\n",
       "      <td>2.472243</td>\n",
       "      <td>2.486333</td>\n",
       "      <td>2.426567</td>\n",
       "      <td>2.390807</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HOT5_F3</th>\n",
       "      <td>0.171875</td>\n",
       "      <td>0.107108</td>\n",
       "      <td>0.106181</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.140698</td>\n",
       "      <td>2.475881</td>\n",
       "      <td>2.489971</td>\n",
       "      <td>2.430205</td>\n",
       "      <td>2.394445</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HOT5_C3</th>\n",
       "      <td>0.202754</td>\n",
       "      <td>0.137987</td>\n",
       "      <td>0.137060</td>\n",
       "      <td>0.140698</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.474515</td>\n",
       "      <td>2.488605</td>\n",
       "      <td>2.428839</td>\n",
       "      <td>2.393078</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HOT1A3</th>\n",
       "      <td>2.537937</td>\n",
       "      <td>2.473170</td>\n",
       "      <td>2.472243</td>\n",
       "      <td>2.475881</td>\n",
       "      <td>2.474515</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.354890</td>\n",
       "      <td>0.614962</td>\n",
       "      <td>0.975858</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CIP</th>\n",
       "      <td>2.552027</td>\n",
       "      <td>2.487260</td>\n",
       "      <td>2.486333</td>\n",
       "      <td>2.489971</td>\n",
       "      <td>2.488605</td>\n",
       "      <td>0.354890</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.629052</td>\n",
       "      <td>0.989948</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25922</th>\n",
       "      <td>2.492261</td>\n",
       "      <td>2.427494</td>\n",
       "      <td>2.426567</td>\n",
       "      <td>2.430205</td>\n",
       "      <td>2.428839</td>\n",
       "      <td>0.614962</td>\n",
       "      <td>0.629052</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.930182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HP15</th>\n",
       "      <td>2.456501</td>\n",
       "      <td>2.391734</td>\n",
       "      <td>2.390807</td>\n",
       "      <td>2.394445</td>\n",
       "      <td>2.393078</td>\n",
       "      <td>0.975858</td>\n",
       "      <td>0.989948</td>\n",
       "      <td>0.930182</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          nitzsch     Pgall     DSS-3   HOT5_F3   HOT5_C3    HOT1A3       CIP  \\\n",
       "nitzsch  0.000000  0.157894  0.164756  0.171875  0.202754  2.537937  2.552027   \n",
       "Pgall    0.157894  0.000000  0.099989  0.107108  0.137987  2.473170  2.487260   \n",
       "DSS-3    0.164756  0.099989  0.000000  0.106181  0.137060  2.472243  2.486333   \n",
       "HOT5_F3  0.171875  0.107108  0.106181  0.000000  0.140698  2.475881  2.489971   \n",
       "HOT5_C3  0.202754  0.137987  0.137060  0.140698  0.000000  2.474515  2.488605   \n",
       "HOT1A3   2.537937  2.473170  2.472243  2.475881  2.474515  0.000000  0.354890   \n",
       "CIP      2.552027  2.487260  2.486333  2.489971  2.488605  0.354890  0.000000   \n",
       "25922    2.492261  2.427494  2.426567  2.430205  2.428839  0.614962  0.629052   \n",
       "HP15     2.456501  2.391734  2.390807  2.394445  2.393078  0.975858  0.989948   \n",
       "\n",
       "            25922      HP15  \n",
       "nitzsch  2.492261  2.456501  \n",
       "Pgall    2.427494  2.391734  \n",
       "DSS-3    2.426567  2.390807  \n",
       "HOT5_F3  2.430205  2.394445  \n",
       "HOT5_C3  2.428839  2.393078  \n",
       "HOT1A3   0.614962  0.975858  \n",
       "CIP      0.629052  0.989948  \n",
       "25922    0.000000  0.930182  \n",
       "HP15     0.930182  0.000000  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phylo_matrix_df = pd.read_csv( 'phylodata/gtdbtk_r220_cophen_dist.csv', index_col=0)\n",
    "phylo_matrix_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "5e742eea-fdc7-4734-9994-5be0c9c3c578",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>strain.nickname</th>\n",
       "      <th>sedthio</th>\n",
       "      <th>kocur</th>\n",
       "      <th>BS11</th>\n",
       "      <th>DE1</th>\n",
       "      <th>plano</th>\n",
       "      <th>serr</th>\n",
       "      <th>crono</th>\n",
       "      <th>Phalo</th>\n",
       "      <th>citrea</th>\n",
       "      <th>...</th>\n",
       "      <th>abie</th>\n",
       "      <th>viet</th>\n",
       "      <th>owen</th>\n",
       "      <th>dies</th>\n",
       "      <th>HOT5_B8</th>\n",
       "      <th>HOT5_B10</th>\n",
       "      <th>HOT5_C3</th>\n",
       "      <th>HOT5_E6</th>\n",
       "      <th>HOT5_F3</th>\n",
       "      <th>PR1red</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>sedthio</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.785921</td>\n",
       "      <td>0.610862</td>\n",
       "      <td>0.619963</td>\n",
       "      <td>0.763889</td>\n",
       "      <td>0.668842</td>\n",
       "      <td>0.687560</td>\n",
       "      <td>0.684987</td>\n",
       "      <td>0.616764</td>\n",
       "      <td>...</td>\n",
       "      <td>0.634877</td>\n",
       "      <td>0.736086</td>\n",
       "      <td>0.724700</td>\n",
       "      <td>0.575476</td>\n",
       "      <td>0.668664</td>\n",
       "      <td>0.699464</td>\n",
       "      <td>0.672104</td>\n",
       "      <td>0.699806</td>\n",
       "      <td>0.700030</td>\n",
       "      <td>0.735570</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>kocur</td>\n",
       "      <td>0.785921</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.778760</td>\n",
       "      <td>0.784421</td>\n",
       "      <td>0.730689</td>\n",
       "      <td>0.790631</td>\n",
       "      <td>0.795322</td>\n",
       "      <td>0.739224</td>\n",
       "      <td>0.793577</td>\n",
       "      <td>...</td>\n",
       "      <td>0.773577</td>\n",
       "      <td>0.759162</td>\n",
       "      <td>0.767815</td>\n",
       "      <td>0.773999</td>\n",
       "      <td>0.768553</td>\n",
       "      <td>0.762369</td>\n",
       "      <td>0.768734</td>\n",
       "      <td>0.762428</td>\n",
       "      <td>0.765255</td>\n",
       "      <td>0.762224</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BS11</td>\n",
       "      <td>0.610862</td>\n",
       "      <td>0.778760</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.231930</td>\n",
       "      <td>0.764562</td>\n",
       "      <td>0.638997</td>\n",
       "      <td>0.622852</td>\n",
       "      <td>0.706065</td>\n",
       "      <td>0.449253</td>\n",
       "      <td>...</td>\n",
       "      <td>0.625240</td>\n",
       "      <td>0.696365</td>\n",
       "      <td>0.708648</td>\n",
       "      <td>0.592007</td>\n",
       "      <td>0.707359</td>\n",
       "      <td>0.711811</td>\n",
       "      <td>0.703874</td>\n",
       "      <td>0.713275</td>\n",
       "      <td>0.715204</td>\n",
       "      <td>0.695376</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DE1</td>\n",
       "      <td>0.619963</td>\n",
       "      <td>0.784421</td>\n",
       "      <td>0.231930</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.767803</td>\n",
       "      <td>0.642565</td>\n",
       "      <td>0.632099</td>\n",
       "      <td>0.713992</td>\n",
       "      <td>0.475689</td>\n",
       "      <td>...</td>\n",
       "      <td>0.622779</td>\n",
       "      <td>0.705771</td>\n",
       "      <td>0.715514</td>\n",
       "      <td>0.599175</td>\n",
       "      <td>0.710065</td>\n",
       "      <td>0.710366</td>\n",
       "      <td>0.709478</td>\n",
       "      <td>0.711435</td>\n",
       "      <td>0.714066</td>\n",
       "      <td>0.705656</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>plano</td>\n",
       "      <td>0.763889</td>\n",
       "      <td>0.730689</td>\n",
       "      <td>0.764562</td>\n",
       "      <td>0.767803</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.787693</td>\n",
       "      <td>0.787455</td>\n",
       "      <td>0.748579</td>\n",
       "      <td>0.768187</td>\n",
       "      <td>...</td>\n",
       "      <td>0.788275</td>\n",
       "      <td>0.743430</td>\n",
       "      <td>0.735257</td>\n",
       "      <td>0.772019</td>\n",
       "      <td>0.772059</td>\n",
       "      <td>0.776212</td>\n",
       "      <td>0.766990</td>\n",
       "      <td>0.773925</td>\n",
       "      <td>0.778300</td>\n",
       "      <td>0.735955</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>HOT5_B10</td>\n",
       "      <td>0.699464</td>\n",
       "      <td>0.762369</td>\n",
       "      <td>0.711811</td>\n",
       "      <td>0.710366</td>\n",
       "      <td>0.776212</td>\n",
       "      <td>0.722249</td>\n",
       "      <td>0.744597</td>\n",
       "      <td>0.572399</td>\n",
       "      <td>0.735951</td>\n",
       "      <td>...</td>\n",
       "      <td>0.672655</td>\n",
       "      <td>0.761654</td>\n",
       "      <td>0.783091</td>\n",
       "      <td>0.684515</td>\n",
       "      <td>0.514578</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.520000</td>\n",
       "      <td>0.039264</td>\n",
       "      <td>0.056862</td>\n",
       "      <td>0.769339</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>HOT5_C3</td>\n",
       "      <td>0.672104</td>\n",
       "      <td>0.768734</td>\n",
       "      <td>0.703874</td>\n",
       "      <td>0.709478</td>\n",
       "      <td>0.766990</td>\n",
       "      <td>0.743128</td>\n",
       "      <td>0.745179</td>\n",
       "      <td>0.592765</td>\n",
       "      <td>0.722528</td>\n",
       "      <td>...</td>\n",
       "      <td>0.678965</td>\n",
       "      <td>0.745319</td>\n",
       "      <td>0.753355</td>\n",
       "      <td>0.677798</td>\n",
       "      <td>0.173502</td>\n",
       "      <td>0.520000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.519369</td>\n",
       "      <td>0.520339</td>\n",
       "      <td>0.757031</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>HOT5_E6</td>\n",
       "      <td>0.699806</td>\n",
       "      <td>0.762428</td>\n",
       "      <td>0.713275</td>\n",
       "      <td>0.711435</td>\n",
       "      <td>0.773925</td>\n",
       "      <td>0.722868</td>\n",
       "      <td>0.744921</td>\n",
       "      <td>0.574010</td>\n",
       "      <td>0.737044</td>\n",
       "      <td>...</td>\n",
       "      <td>0.673075</td>\n",
       "      <td>0.763439</td>\n",
       "      <td>0.785005</td>\n",
       "      <td>0.683331</td>\n",
       "      <td>0.516426</td>\n",
       "      <td>0.039264</td>\n",
       "      <td>0.519369</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.055768</td>\n",
       "      <td>0.772102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>HOT5_F3</td>\n",
       "      <td>0.700030</td>\n",
       "      <td>0.765255</td>\n",
       "      <td>0.715204</td>\n",
       "      <td>0.714066</td>\n",
       "      <td>0.778300</td>\n",
       "      <td>0.720696</td>\n",
       "      <td>0.745500</td>\n",
       "      <td>0.578396</td>\n",
       "      <td>0.737984</td>\n",
       "      <td>...</td>\n",
       "      <td>0.673148</td>\n",
       "      <td>0.762467</td>\n",
       "      <td>0.785340</td>\n",
       "      <td>0.686090</td>\n",
       "      <td>0.516440</td>\n",
       "      <td>0.056862</td>\n",
       "      <td>0.520339</td>\n",
       "      <td>0.055768</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.772369</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>PR1red</td>\n",
       "      <td>0.735570</td>\n",
       "      <td>0.762224</td>\n",
       "      <td>0.695376</td>\n",
       "      <td>0.705656</td>\n",
       "      <td>0.735955</td>\n",
       "      <td>0.782785</td>\n",
       "      <td>0.780665</td>\n",
       "      <td>0.750000</td>\n",
       "      <td>0.707242</td>\n",
       "      <td>...</td>\n",
       "      <td>0.763791</td>\n",
       "      <td>0.401173</td>\n",
       "      <td>0.523610</td>\n",
       "      <td>0.743539</td>\n",
       "      <td>0.756663</td>\n",
       "      <td>0.769339</td>\n",
       "      <td>0.757031</td>\n",
       "      <td>0.772102</td>\n",
       "      <td>0.772369</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>88 rows × 89 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   strain.nickname   sedthio     kocur      BS11       DE1     plano  \\\n",
       "0          sedthio  0.000000  0.785921  0.610862  0.619963  0.763889   \n",
       "1            kocur  0.785921  0.000000  0.778760  0.784421  0.730689   \n",
       "2             BS11  0.610862  0.778760  0.000000  0.231930  0.764562   \n",
       "3              DE1  0.619963  0.784421  0.231930  0.000000  0.767803   \n",
       "4            plano  0.763889  0.730689  0.764562  0.767803  0.000000   \n",
       "..             ...       ...       ...       ...       ...       ...   \n",
       "83        HOT5_B10  0.699464  0.762369  0.711811  0.710366  0.776212   \n",
       "84         HOT5_C3  0.672104  0.768734  0.703874  0.709478  0.766990   \n",
       "85         HOT5_E6  0.699806  0.762428  0.713275  0.711435  0.773925   \n",
       "86         HOT5_F3  0.700030  0.765255  0.715204  0.714066  0.778300   \n",
       "87          PR1red  0.735570  0.762224  0.695376  0.705656  0.735955   \n",
       "\n",
       "        serr     crono     Phalo    citrea  ...      abie      viet      owen  \\\n",
       "0   0.668842  0.687560  0.684987  0.616764  ...  0.634877  0.736086  0.724700   \n",
       "1   0.790631  0.795322  0.739224  0.793577  ...  0.773577  0.759162  0.767815   \n",
       "2   0.638997  0.622852  0.706065  0.449253  ...  0.625240  0.696365  0.708648   \n",
       "3   0.642565  0.632099  0.713992  0.475689  ...  0.622779  0.705771  0.715514   \n",
       "4   0.787693  0.787455  0.748579  0.768187  ...  0.788275  0.743430  0.735257   \n",
       "..       ...       ...       ...       ...  ...       ...       ...       ...   \n",
       "83  0.722249  0.744597  0.572399  0.735951  ...  0.672655  0.761654  0.783091   \n",
       "84  0.743128  0.745179  0.592765  0.722528  ...  0.678965  0.745319  0.753355   \n",
       "85  0.722868  0.744921  0.574010  0.737044  ...  0.673075  0.763439  0.785005   \n",
       "86  0.720696  0.745500  0.578396  0.737984  ...  0.673148  0.762467  0.785340   \n",
       "87  0.782785  0.780665  0.750000  0.707242  ...  0.763791  0.401173  0.523610   \n",
       "\n",
       "        dies   HOT5_B8  HOT5_B10   HOT5_C3   HOT5_E6   HOT5_F3    PR1red  \n",
       "0   0.575476  0.668664  0.699464  0.672104  0.699806  0.700030  0.735570  \n",
       "1   0.773999  0.768553  0.762369  0.768734  0.762428  0.765255  0.762224  \n",
       "2   0.592007  0.707359  0.711811  0.703874  0.713275  0.715204  0.695376  \n",
       "3   0.599175  0.710065  0.710366  0.709478  0.711435  0.714066  0.705656  \n",
       "4   0.772019  0.772059  0.776212  0.766990  0.773925  0.778300  0.735955  \n",
       "..       ...       ...       ...       ...       ...       ...       ...  \n",
       "83  0.684515  0.514578  0.000000  0.520000  0.039264  0.056862  0.769339  \n",
       "84  0.677798  0.173502  0.520000  0.000000  0.519369  0.520339  0.757031  \n",
       "85  0.683331  0.516426  0.039264  0.519369  0.000000  0.055768  0.772102  \n",
       "86  0.686090  0.516440  0.056862  0.520339  0.055768  0.000000  0.772369  \n",
       "87  0.743539  0.756663  0.769339  0.757031  0.772102  0.772369  0.000000  \n",
       "\n",
       "[88 rows x 89 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_matrix_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "410e191b-845a-4e40-887b-1c27f32934d2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Group</th>\n",
       "      <th>Name</th>\n",
       "      <th>NCBI taxonomy id</th>\n",
       "      <th>strain_nickname</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Strong</td>\n",
       "      <td>Alteromonas macleodii</td>\n",
       "      <td>28108</td>\n",
       "      <td>HOT1A3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Strong</td>\n",
       "      <td>Pseudoalteromonas haloplanktis</td>\n",
       "      <td>326442</td>\n",
       "      <td>CIP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Sustained</td>\n",
       "      <td>Ruegeria pomeroyi</td>\n",
       "      <td>246200</td>\n",
       "      <td>DSS-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Sustained</td>\n",
       "      <td>Sulfitobacter pseudonitzschiae</td>\n",
       "      <td>1402135</td>\n",
       "      <td>nitzsch</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Weak</td>\n",
       "      <td>Marinovum HOT5_F3</td>\n",
       "      <td>367771</td>\n",
       "      <td>HOT5_F3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Inhibited</td>\n",
       "      <td>Marinobacter adhaerens</td>\n",
       "      <td>225937</td>\n",
       "      <td>HP15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Weak</td>\n",
       "      <td>Roseovarius HOT5_C3</td>\n",
       "      <td>215743</td>\n",
       "      <td>HOT5_C3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Inhibited</td>\n",
       "      <td>Phaeobacter gallaeciensis</td>\n",
       "      <td>1423144</td>\n",
       "      <td>Pgall</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Group                            Name  NCBI taxonomy id strain_nickname\n",
       "0     Strong           Alteromonas macleodii             28108          HOT1A3\n",
       "1     Strong  Pseudoalteromonas haloplanktis            326442             CIP\n",
       "2  Sustained               Ruegeria pomeroyi            246200           DSS-3\n",
       "3  Sustained  Sulfitobacter pseudonitzschiae           1402135         nitzsch\n",
       "5       Weak               Marinovum HOT5_F3            367771         HOT5_F3\n",
       "6  Inhibited          Marinobacter adhaerens            225937            HP15\n",
       "7       Weak             Roseovarius HOT5_C3            215743         HOT5_C3\n",
       "8  Inhibited       Phaeobacter gallaeciensis           1423144           Pgall"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "d19f72a9-1c30-41bd-a221-3eebee1c362a",
   "metadata": {},
   "outputs": [],
   "source": [
    "small_pairwise_dist_df = pairwise_dist_df.loc[\n",
    "    pairwise_dist_df.strain1.isin(metadata_df.strain_nickname) &\n",
    "    pairwise_dist_df.strain2.isin(metadata_df.strain_nickname) \n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "a8062f0f-f660-4ead-9885-b586299b9838",
   "metadata": {},
   "outputs": [],
   "source": [
    "phylo_dist_matrix = small_pairwise_dist_df.pivot(index='strain1', columns='strain2')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d0626b9a-c29b-4927-ae74-28dce2b4a278",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "4c599e55-72e2-4713-9928-dc6b3fe4953f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HOT1A3</th>\n",
       "      <th>CIP</th>\n",
       "      <th>DSS-3</th>\n",
       "      <th>nitzsch</th>\n",
       "      <th>HOT5_F3</th>\n",
       "      <th>HP15</th>\n",
       "      <th>HOT5_C3</th>\n",
       "      <th>Pgall</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>strain_nickname</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HOT1A3</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.447004</td>\n",
       "      <td>0.705086</td>\n",
       "      <td>0.702623</td>\n",
       "      <td>0.714925</td>\n",
       "      <td>0.564989</td>\n",
       "      <td>0.704844</td>\n",
       "      <td>0.689370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CIP</th>\n",
       "      <td>0.447004</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.713417</td>\n",
       "      <td>0.718618</td>\n",
       "      <td>0.729038</td>\n",
       "      <td>0.572837</td>\n",
       "      <td>0.722444</td>\n",
       "      <td>0.696241</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DSS-3</th>\n",
       "      <td>0.705086</td>\n",
       "      <td>0.713417</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.428622</td>\n",
       "      <td>0.492586</td>\n",
       "      <td>0.659122</td>\n",
       "      <td>0.466273</td>\n",
       "      <td>0.392661</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>nitzsch</th>\n",
       "      <td>0.702623</td>\n",
       "      <td>0.718618</td>\n",
       "      <td>0.428622</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.458914</td>\n",
       "      <td>0.662634</td>\n",
       "      <td>0.462587</td>\n",
       "      <td>0.436338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HOT5_F3</th>\n",
       "      <td>0.714925</td>\n",
       "      <td>0.729038</td>\n",
       "      <td>0.492586</td>\n",
       "      <td>0.458914</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.678744</td>\n",
       "      <td>0.520339</td>\n",
       "      <td>0.470034</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HP15</th>\n",
       "      <td>0.564989</td>\n",
       "      <td>0.572837</td>\n",
       "      <td>0.659122</td>\n",
       "      <td>0.662634</td>\n",
       "      <td>0.678744</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.656250</td>\n",
       "      <td>0.649961</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HOT5_C3</th>\n",
       "      <td>0.704844</td>\n",
       "      <td>0.722444</td>\n",
       "      <td>0.466273</td>\n",
       "      <td>0.462587</td>\n",
       "      <td>0.520339</td>\n",
       "      <td>0.656250</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.455786</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pgall</th>\n",
       "      <td>0.689370</td>\n",
       "      <td>0.696241</td>\n",
       "      <td>0.392661</td>\n",
       "      <td>0.436338</td>\n",
       "      <td>0.470034</td>\n",
       "      <td>0.649961</td>\n",
       "      <td>0.455786</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   HOT1A3       CIP     DSS-3   nitzsch   HOT5_F3      HP15  \\\n",
       "strain_nickname                                                               \n",
       "HOT1A3           0.000000  0.447004  0.705086  0.702623  0.714925  0.564989   \n",
       "CIP              0.447004  0.000000  0.713417  0.718618  0.729038  0.572837   \n",
       "DSS-3            0.705086  0.713417  0.000000  0.428622  0.492586  0.659122   \n",
       "nitzsch          0.702623  0.718618  0.428622  0.000000  0.458914  0.662634   \n",
       "HOT5_F3          0.714925  0.729038  0.492586  0.458914  0.000000  0.678744   \n",
       "HP15             0.564989  0.572837  0.659122  0.662634  0.678744  0.000000   \n",
       "HOT5_C3          0.704844  0.722444  0.466273  0.462587  0.520339  0.656250   \n",
       "Pgall            0.689370  0.696241  0.392661  0.436338  0.470034  0.649961   \n",
       "\n",
       "                  HOT5_C3     Pgall  \n",
       "strain_nickname                      \n",
       "HOT1A3           0.704844  0.689370  \n",
       "CIP              0.722444  0.696241  \n",
       "DSS-3            0.466273  0.392661  \n",
       "nitzsch          0.462587  0.436338  \n",
       "HOT5_F3          0.520339  0.470034  \n",
       "HP15             0.656250  0.649961  \n",
       "HOT5_C3          0.000000  0.455786  \n",
       "Pgall            0.455786  0.000000  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_dist_matrix = gene_matrix_df.loc[gene_matrix_df['strain.nickname'].isin(metadata_df.strain_nickname)]\n",
    "gene_dist_matrix.index = gene_dist_matrix['strain.nickname']\n",
    "gene_dist_matrix =  gene_dist_matrix[metadata_df.strain_nickname]\n",
    "gene_dist_matrix = gene_dist_matrix.reindex(index=metadata_df.strain_nickname)\n",
    "gene_dist_matrix\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fa5858d4-0145-4623-8c7a-0824c42d3a4d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "255301bc-6d7e-4f3c-bfa0-fd9997519751",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>strain</th>\n",
       "      <th>sedthio</th>\n",
       "      <th>kocur</th>\n",
       "      <th>BS11</th>\n",
       "      <th>DE1</th>\n",
       "      <th>plano</th>\n",
       "      <th>serr</th>\n",
       "      <th>crono</th>\n",
       "      <th>Phalo</th>\n",
       "      <th>citrea</th>\n",
       "      <th>...</th>\n",
       "      <th>abie</th>\n",
       "      <th>viet</th>\n",
       "      <th>owen</th>\n",
       "      <th>dies</th>\n",
       "      <th>HOT5_B8</th>\n",
       "      <th>HOT5_B10</th>\n",
       "      <th>HOT5_C3</th>\n",
       "      <th>HOT5_E6</th>\n",
       "      <th>HOT5_F3</th>\n",
       "      <th>PR1red</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>sedthio</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.577181</td>\n",
       "      <td>0.370861</td>\n",
       "      <td>0.353741</td>\n",
       "      <td>0.550336</td>\n",
       "      <td>0.484694</td>\n",
       "      <td>0.467742</td>\n",
       "      <td>0.512500</td>\n",
       "      <td>0.376712</td>\n",
       "      <td>...</td>\n",
       "      <td>0.409091</td>\n",
       "      <td>0.468966</td>\n",
       "      <td>0.481203</td>\n",
       "      <td>0.326389</td>\n",
       "      <td>0.386207</td>\n",
       "      <td>0.494186</td>\n",
       "      <td>0.386207</td>\n",
       "      <td>0.505814</td>\n",
       "      <td>0.488372</td>\n",
       "      <td>0.492857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>kocur</td>\n",
       "      <td>0.577181</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.478873</td>\n",
       "      <td>0.496454</td>\n",
       "      <td>0.475410</td>\n",
       "      <td>0.572193</td>\n",
       "      <td>0.559322</td>\n",
       "      <td>0.514286</td>\n",
       "      <td>0.531915</td>\n",
       "      <td>...</td>\n",
       "      <td>0.529412</td>\n",
       "      <td>0.464000</td>\n",
       "      <td>0.504348</td>\n",
       "      <td>0.593333</td>\n",
       "      <td>0.542857</td>\n",
       "      <td>0.503268</td>\n",
       "      <td>0.532374</td>\n",
       "      <td>0.506579</td>\n",
       "      <td>0.496732</td>\n",
       "      <td>0.452991</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BS11</td>\n",
       "      <td>0.370861</td>\n",
       "      <td>0.478873</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.032000</td>\n",
       "      <td>0.513514</td>\n",
       "      <td>0.372973</td>\n",
       "      <td>0.309942</td>\n",
       "      <td>0.428571</td>\n",
       "      <td>0.201493</td>\n",
       "      <td>...</td>\n",
       "      <td>0.319527</td>\n",
       "      <td>0.350365</td>\n",
       "      <td>0.462687</td>\n",
       "      <td>0.366667</td>\n",
       "      <td>0.434211</td>\n",
       "      <td>0.443787</td>\n",
       "      <td>0.434211</td>\n",
       "      <td>0.464706</td>\n",
       "      <td>0.437870</td>\n",
       "      <td>0.395522</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DE1</td>\n",
       "      <td>0.353741</td>\n",
       "      <td>0.496454</td>\n",
       "      <td>0.032000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.520548</td>\n",
       "      <td>0.394595</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>0.454545</td>\n",
       "      <td>0.204545</td>\n",
       "      <td>...</td>\n",
       "      <td>0.343195</td>\n",
       "      <td>0.367647</td>\n",
       "      <td>0.458015</td>\n",
       "      <td>0.349315</td>\n",
       "      <td>0.418919</td>\n",
       "      <td>0.467456</td>\n",
       "      <td>0.418919</td>\n",
       "      <td>0.488235</td>\n",
       "      <td>0.461538</td>\n",
       "      <td>0.413534</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>plano</td>\n",
       "      <td>0.550336</td>\n",
       "      <td>0.475410</td>\n",
       "      <td>0.513514</td>\n",
       "      <td>0.520548</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.595855</td>\n",
       "      <td>0.553073</td>\n",
       "      <td>0.485714</td>\n",
       "      <td>0.503546</td>\n",
       "      <td>...</td>\n",
       "      <td>0.564972</td>\n",
       "      <td>0.492308</td>\n",
       "      <td>0.508475</td>\n",
       "      <td>0.547297</td>\n",
       "      <td>0.514286</td>\n",
       "      <td>0.525316</td>\n",
       "      <td>0.514286</td>\n",
       "      <td>0.528662</td>\n",
       "      <td>0.528302</td>\n",
       "      <td>0.495935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>HOT5_B10</td>\n",
       "      <td>0.494186</td>\n",
       "      <td>0.503268</td>\n",
       "      <td>0.443787</td>\n",
       "      <td>0.467456</td>\n",
       "      <td>0.525316</td>\n",
       "      <td>0.490385</td>\n",
       "      <td>0.459184</td>\n",
       "      <td>0.300000</td>\n",
       "      <td>0.523256</td>\n",
       "      <td>...</td>\n",
       "      <td>0.428571</td>\n",
       "      <td>0.477707</td>\n",
       "      <td>0.530201</td>\n",
       "      <td>0.517241</td>\n",
       "      <td>0.292517</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.292517</td>\n",
       "      <td>0.028777</td>\n",
       "      <td>0.035461</td>\n",
       "      <td>0.480000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>HOT5_C3</td>\n",
       "      <td>0.386207</td>\n",
       "      <td>0.532374</td>\n",
       "      <td>0.434211</td>\n",
       "      <td>0.418919</td>\n",
       "      <td>0.514286</td>\n",
       "      <td>0.540404</td>\n",
       "      <td>0.494565</td>\n",
       "      <td>0.357143</td>\n",
       "      <td>0.503268</td>\n",
       "      <td>...</td>\n",
       "      <td>0.463277</td>\n",
       "      <td>0.460432</td>\n",
       "      <td>0.472441</td>\n",
       "      <td>0.436242</td>\n",
       "      <td>0.017544</td>\n",
       "      <td>0.292517</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.294521</td>\n",
       "      <td>0.285714</td>\n",
       "      <td>0.462121</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>HOT5_E6</td>\n",
       "      <td>0.505814</td>\n",
       "      <td>0.506579</td>\n",
       "      <td>0.464706</td>\n",
       "      <td>0.488235</td>\n",
       "      <td>0.528662</td>\n",
       "      <td>0.507177</td>\n",
       "      <td>0.477157</td>\n",
       "      <td>0.302013</td>\n",
       "      <td>0.543353</td>\n",
       "      <td>...</td>\n",
       "      <td>0.439153</td>\n",
       "      <td>0.490446</td>\n",
       "      <td>0.553333</td>\n",
       "      <td>0.528736</td>\n",
       "      <td>0.294521</td>\n",
       "      <td>0.028777</td>\n",
       "      <td>0.294521</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.035714</td>\n",
       "      <td>0.493333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>HOT5_F3</td>\n",
       "      <td>0.488372</td>\n",
       "      <td>0.496732</td>\n",
       "      <td>0.437870</td>\n",
       "      <td>0.461538</td>\n",
       "      <td>0.528302</td>\n",
       "      <td>0.485577</td>\n",
       "      <td>0.454082</td>\n",
       "      <td>0.315789</td>\n",
       "      <td>0.517442</td>\n",
       "      <td>...</td>\n",
       "      <td>0.414894</td>\n",
       "      <td>0.461538</td>\n",
       "      <td>0.543046</td>\n",
       "      <td>0.511494</td>\n",
       "      <td>0.285714</td>\n",
       "      <td>0.035461</td>\n",
       "      <td>0.285714</td>\n",
       "      <td>0.035714</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.473333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>PR1red</td>\n",
       "      <td>0.492857</td>\n",
       "      <td>0.452991</td>\n",
       "      <td>0.395522</td>\n",
       "      <td>0.413534</td>\n",
       "      <td>0.495935</td>\n",
       "      <td>0.569892</td>\n",
       "      <td>0.531792</td>\n",
       "      <td>0.443609</td>\n",
       "      <td>0.427481</td>\n",
       "      <td>...</td>\n",
       "      <td>0.517857</td>\n",
       "      <td>0.163462</td>\n",
       "      <td>0.272727</td>\n",
       "      <td>0.510638</td>\n",
       "      <td>0.462121</td>\n",
       "      <td>0.480000</td>\n",
       "      <td>0.462121</td>\n",
       "      <td>0.493333</td>\n",
       "      <td>0.473333</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>88 rows × 89 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      strain   sedthio     kocur      BS11       DE1     plano      serr  \\\n",
       "0    sedthio  0.000000  0.577181  0.370861  0.353741  0.550336  0.484694   \n",
       "1      kocur  0.577181  0.000000  0.478873  0.496454  0.475410  0.572193   \n",
       "2       BS11  0.370861  0.478873  0.000000  0.032000  0.513514  0.372973   \n",
       "3        DE1  0.353741  0.496454  0.032000  0.000000  0.520548  0.394595   \n",
       "4      plano  0.550336  0.475410  0.513514  0.520548  0.000000  0.595855   \n",
       "..       ...       ...       ...       ...       ...       ...       ...   \n",
       "83  HOT5_B10  0.494186  0.503268  0.443787  0.467456  0.525316  0.490385   \n",
       "84   HOT5_C3  0.386207  0.532374  0.434211  0.418919  0.514286  0.540404   \n",
       "85   HOT5_E6  0.505814  0.506579  0.464706  0.488235  0.528662  0.507177   \n",
       "86   HOT5_F3  0.488372  0.496732  0.437870  0.461538  0.528302  0.485577   \n",
       "87    PR1red  0.492857  0.452991  0.395522  0.413534  0.495935  0.569892   \n",
       "\n",
       "       crono     Phalo    citrea  ...      abie      viet      owen      dies  \\\n",
       "0   0.467742  0.512500  0.376712  ...  0.409091  0.468966  0.481203  0.326389   \n",
       "1   0.559322  0.514286  0.531915  ...  0.529412  0.464000  0.504348  0.593333   \n",
       "2   0.309942  0.428571  0.201493  ...  0.319527  0.350365  0.462687  0.366667   \n",
       "3   0.333333  0.454545  0.204545  ...  0.343195  0.367647  0.458015  0.349315   \n",
       "4   0.553073  0.485714  0.503546  ...  0.564972  0.492308  0.508475  0.547297   \n",
       "..       ...       ...       ...  ...       ...       ...       ...       ...   \n",
       "83  0.459184  0.300000  0.523256  ...  0.428571  0.477707  0.530201  0.517241   \n",
       "84  0.494565  0.357143  0.503268  ...  0.463277  0.460432  0.472441  0.436242   \n",
       "85  0.477157  0.302013  0.543353  ...  0.439153  0.490446  0.553333  0.528736   \n",
       "86  0.454082  0.315789  0.517442  ...  0.414894  0.461538  0.543046  0.511494   \n",
       "87  0.531792  0.443609  0.427481  ...  0.517857  0.163462  0.272727  0.510638   \n",
       "\n",
       "     HOT5_B8  HOT5_B10   HOT5_C3   HOT5_E6   HOT5_F3    PR1red  \n",
       "0   0.386207  0.494186  0.386207  0.505814  0.488372  0.492857  \n",
       "1   0.542857  0.503268  0.532374  0.506579  0.496732  0.452991  \n",
       "2   0.434211  0.443787  0.434211  0.464706  0.437870  0.395522  \n",
       "3   0.418919  0.467456  0.418919  0.488235  0.461538  0.413534  \n",
       "4   0.514286  0.525316  0.514286  0.528662  0.528302  0.495935  \n",
       "..       ...       ...       ...       ...       ...       ...  \n",
       "83  0.292517  0.000000  0.292517  0.028777  0.035461  0.480000  \n",
       "84  0.017544  0.292517  0.000000  0.294521  0.285714  0.462121  \n",
       "85  0.294521  0.028777  0.294521  0.000000  0.035714  0.493333  \n",
       "86  0.285714  0.035461  0.285714  0.035714  0.000000  0.473333  \n",
       "87  0.462121  0.480000  0.462121  0.493333  0.473333  0.000000  \n",
       "\n",
       "[88 rows x 89 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trait_matrix_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "104d8420-16ac-4d07-ba27-773be5e446a0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HOT1A3</th>\n",
       "      <th>CIP</th>\n",
       "      <th>DSS-3</th>\n",
       "      <th>nitzsch</th>\n",
       "      <th>HOT5_F3</th>\n",
       "      <th>HP15</th>\n",
       "      <th>HOT5_C3</th>\n",
       "      <th>Pgall</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>strain_nickname</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HOT1A3</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.208955</td>\n",
       "      <td>0.432258</td>\n",
       "      <td>0.405229</td>\n",
       "      <td>0.437870</td>\n",
       "      <td>0.324138</td>\n",
       "      <td>0.423841</td>\n",
       "      <td>0.411392</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CIP</th>\n",
       "      <td>0.208955</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.477124</td>\n",
       "      <td>0.450331</td>\n",
       "      <td>0.497041</td>\n",
       "      <td>0.359155</td>\n",
       "      <td>0.480000</td>\n",
       "      <td>0.424837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DSS-3</th>\n",
       "      <td>0.432258</td>\n",
       "      <td>0.477124</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.218045</td>\n",
       "      <td>0.298013</td>\n",
       "      <td>0.405405</td>\n",
       "      <td>0.195312</td>\n",
       "      <td>0.218978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>nitzsch</th>\n",
       "      <td>0.405229</td>\n",
       "      <td>0.450331</td>\n",
       "      <td>0.218045</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.232877</td>\n",
       "      <td>0.398649</td>\n",
       "      <td>0.201550</td>\n",
       "      <td>0.198529</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HOT5_F3</th>\n",
       "      <td>0.437870</td>\n",
       "      <td>0.497041</td>\n",
       "      <td>0.298013</td>\n",
       "      <td>0.232877</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.383648</td>\n",
       "      <td>0.285714</td>\n",
       "      <td>0.267974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HP15</th>\n",
       "      <td>0.324138</td>\n",
       "      <td>0.359155</td>\n",
       "      <td>0.405405</td>\n",
       "      <td>0.398649</td>\n",
       "      <td>0.383648</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.373239</td>\n",
       "      <td>0.373333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HOT5_C3</th>\n",
       "      <td>0.423841</td>\n",
       "      <td>0.480000</td>\n",
       "      <td>0.195312</td>\n",
       "      <td>0.201550</td>\n",
       "      <td>0.285714</td>\n",
       "      <td>0.373239</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.216418</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pgall</th>\n",
       "      <td>0.411392</td>\n",
       "      <td>0.424837</td>\n",
       "      <td>0.218978</td>\n",
       "      <td>0.198529</td>\n",
       "      <td>0.267974</td>\n",
       "      <td>0.373333</td>\n",
       "      <td>0.216418</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   HOT1A3       CIP     DSS-3   nitzsch   HOT5_F3      HP15  \\\n",
       "strain_nickname                                                               \n",
       "HOT1A3           0.000000  0.208955  0.432258  0.405229  0.437870  0.324138   \n",
       "CIP              0.208955  0.000000  0.477124  0.450331  0.497041  0.359155   \n",
       "DSS-3            0.432258  0.477124  0.000000  0.218045  0.298013  0.405405   \n",
       "nitzsch          0.405229  0.450331  0.218045  0.000000  0.232877  0.398649   \n",
       "HOT5_F3          0.437870  0.497041  0.298013  0.232877  0.000000  0.383648   \n",
       "HP15             0.324138  0.359155  0.405405  0.398649  0.383648  0.000000   \n",
       "HOT5_C3          0.423841  0.480000  0.195312  0.201550  0.285714  0.373239   \n",
       "Pgall            0.411392  0.424837  0.218978  0.198529  0.267974  0.373333   \n",
       "\n",
       "                  HOT5_C3     Pgall  \n",
       "strain_nickname                      \n",
       "HOT1A3           0.423841  0.411392  \n",
       "CIP              0.480000  0.424837  \n",
       "DSS-3            0.195312  0.218978  \n",
       "nitzsch          0.201550  0.198529  \n",
       "HOT5_F3          0.285714  0.267974  \n",
       "HP15             0.373239  0.373333  \n",
       "HOT5_C3          0.000000  0.216418  \n",
       "Pgall            0.216418  0.000000  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trait_dist_matrix = trait_matrix_df.loc[trait_matrix_df['strain'].isin(metadata_df.strain_nickname)]\n",
    "trait_dist_matrix.index = trait_dist_matrix['strain']\n",
    "trait_dist_matrix =  trait_dist_matrix[metadata_df.strain_nickname]\n",
    "trait_dist_matrix = trait_dist_matrix.reindex(index=metadata_df.strain_nickname)\n",
    "trait_dist_matrix\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "15547726-62a7-4186-8d0f-618c1e2e92b4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HOT1A3</th>\n",
       "      <th>CIP</th>\n",
       "      <th>DSS-3</th>\n",
       "      <th>nitzsch</th>\n",
       "      <th>HOT5_F3</th>\n",
       "      <th>HP15</th>\n",
       "      <th>HOT5_C3</th>\n",
       "      <th>Pgall</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>strain_nickname</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HOT1A3</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.447004</td>\n",
       "      <td>0.705086</td>\n",
       "      <td>0.702623</td>\n",
       "      <td>0.714925</td>\n",
       "      <td>0.564989</td>\n",
       "      <td>0.704844</td>\n",
       "      <td>0.689370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CIP</th>\n",
       "      <td>0.447004</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.713417</td>\n",
       "      <td>0.718618</td>\n",
       "      <td>0.729038</td>\n",
       "      <td>0.572837</td>\n",
       "      <td>0.722444</td>\n",
       "      <td>0.696241</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DSS-3</th>\n",
       "      <td>0.705086</td>\n",
       "      <td>0.713417</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.428622</td>\n",
       "      <td>0.492586</td>\n",
       "      <td>0.659122</td>\n",
       "      <td>0.466273</td>\n",
       "      <td>0.392661</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>nitzsch</th>\n",
       "      <td>0.702623</td>\n",
       "      <td>0.718618</td>\n",
       "      <td>0.428622</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.458914</td>\n",
       "      <td>0.662634</td>\n",
       "      <td>0.462587</td>\n",
       "      <td>0.436338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HOT5_F3</th>\n",
       "      <td>0.714925</td>\n",
       "      <td>0.729038</td>\n",
       "      <td>0.492586</td>\n",
       "      <td>0.458914</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.678744</td>\n",
       "      <td>0.520339</td>\n",
       "      <td>0.470034</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HP15</th>\n",
       "      <td>0.564989</td>\n",
       "      <td>0.572837</td>\n",
       "      <td>0.659122</td>\n",
       "      <td>0.662634</td>\n",
       "      <td>0.678744</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.656250</td>\n",
       "      <td>0.649961</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HOT5_C3</th>\n",
       "      <td>0.704844</td>\n",
       "      <td>0.722444</td>\n",
       "      <td>0.466273</td>\n",
       "      <td>0.462587</td>\n",
       "      <td>0.520339</td>\n",
       "      <td>0.656250</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.455786</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pgall</th>\n",
       "      <td>0.689370</td>\n",
       "      <td>0.696241</td>\n",
       "      <td>0.392661</td>\n",
       "      <td>0.436338</td>\n",
       "      <td>0.470034</td>\n",
       "      <td>0.649961</td>\n",
       "      <td>0.455786</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   HOT1A3       CIP     DSS-3   nitzsch   HOT5_F3      HP15  \\\n",
       "strain_nickname                                                               \n",
       "HOT1A3           0.000000  0.447004  0.705086  0.702623  0.714925  0.564989   \n",
       "CIP              0.447004  0.000000  0.713417  0.718618  0.729038  0.572837   \n",
       "DSS-3            0.705086  0.713417  0.000000  0.428622  0.492586  0.659122   \n",
       "nitzsch          0.702623  0.718618  0.428622  0.000000  0.458914  0.662634   \n",
       "HOT5_F3          0.714925  0.729038  0.492586  0.458914  0.000000  0.678744   \n",
       "HP15             0.564989  0.572837  0.659122  0.662634  0.678744  0.000000   \n",
       "HOT5_C3          0.704844  0.722444  0.466273  0.462587  0.520339  0.656250   \n",
       "Pgall            0.689370  0.696241  0.392661  0.436338  0.470034  0.649961   \n",
       "\n",
       "                  HOT5_C3     Pgall  \n",
       "strain_nickname                      \n",
       "HOT1A3           0.704844  0.689370  \n",
       "CIP              0.722444  0.696241  \n",
       "DSS-3            0.466273  0.392661  \n",
       "nitzsch          0.462587  0.436338  \n",
       "HOT5_F3          0.520339  0.470034  \n",
       "HP15             0.656250  0.649961  \n",
       "HOT5_C3          0.000000  0.455786  \n",
       "Pgall            0.455786  0.000000  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_dist_matrix = gene_matrix_df.loc[gene_matrix_df['strain.nickname'].isin(metadata_df.strain_nickname)]\n",
    "gene_dist_matrix.index = gene_dist_matrix['strain.nickname']\n",
    "gene_dist_matrix =  gene_dist_matrix[metadata_df.strain_nickname]\n",
    "gene_dist_matrix = gene_dist_matrix.reindex(index=metadata_df.strain_nickname)\n",
    "gene_dist_matrix\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "fa6ef86a-bb83-4bf8-93a4-7b7d8f6d4841",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Group</th>\n",
       "      <th>Name</th>\n",
       "      <th>NCBI taxonomy id</th>\n",
       "      <th>strain_nickname</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Weak</td>\n",
       "      <td>Marinovum HOT5_F3</td>\n",
       "      <td>367771</td>\n",
       "      <td>HOT5_F3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Weak</td>\n",
       "      <td>Roseovarius HOT5_C3</td>\n",
       "      <td>215743</td>\n",
       "      <td>HOT5_C3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Group                 Name  NCBI taxonomy id strain_nickname\n",
       "5  Weak    Marinovum HOT5_F3            367771         HOT5_F3\n",
       "7  Weak  Roseovarius HOT5_C3            215743         HOT5_C3"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata_df.loc[~metadata_df.strain_nickname.isin(pairwise_dist_df.strain1)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "02c62646-2d2d-40c5-80c0-3e92ead93168",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HOT1A3</th>\n",
       "      <th>CIP</th>\n",
       "      <th>DSS-3</th>\n",
       "      <th>nitzsch</th>\n",
       "      <th>HOT5_F3</th>\n",
       "      <th>HP15</th>\n",
       "      <th>HOT5_C3</th>\n",
       "      <th>Pgall</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>strain_nickname</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HOT1A3</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.354890</td>\n",
       "      <td>2.472243</td>\n",
       "      <td>2.537937</td>\n",
       "      <td>2.475881</td>\n",
       "      <td>0.975858</td>\n",
       "      <td>2.474515</td>\n",
       "      <td>2.473170</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CIP</th>\n",
       "      <td>0.354890</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.486333</td>\n",
       "      <td>2.552027</td>\n",
       "      <td>2.489971</td>\n",
       "      <td>0.989948</td>\n",
       "      <td>2.488605</td>\n",
       "      <td>2.487260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DSS-3</th>\n",
       "      <td>2.472243</td>\n",
       "      <td>2.486333</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.164756</td>\n",
       "      <td>0.106181</td>\n",
       "      <td>2.390807</td>\n",
       "      <td>0.137060</td>\n",
       "      <td>0.099989</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>nitzsch</th>\n",
       "      <td>2.537937</td>\n",
       "      <td>2.552027</td>\n",
       "      <td>0.164756</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.171875</td>\n",
       "      <td>2.456501</td>\n",
       "      <td>0.202754</td>\n",
       "      <td>0.157894</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HOT5_F3</th>\n",
       "      <td>2.475881</td>\n",
       "      <td>2.489971</td>\n",
       "      <td>0.106181</td>\n",
       "      <td>0.171875</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.394445</td>\n",
       "      <td>0.140698</td>\n",
       "      <td>0.107108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HP15</th>\n",
       "      <td>0.975858</td>\n",
       "      <td>0.989948</td>\n",
       "      <td>2.390807</td>\n",
       "      <td>2.456501</td>\n",
       "      <td>2.394445</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.393078</td>\n",
       "      <td>2.391734</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HOT5_C3</th>\n",
       "      <td>2.474515</td>\n",
       "      <td>2.488605</td>\n",
       "      <td>0.137060</td>\n",
       "      <td>0.202754</td>\n",
       "      <td>0.140698</td>\n",
       "      <td>2.393078</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.137987</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pgall</th>\n",
       "      <td>2.473170</td>\n",
       "      <td>2.487260</td>\n",
       "      <td>0.099989</td>\n",
       "      <td>0.157894</td>\n",
       "      <td>0.107108</td>\n",
       "      <td>2.391734</td>\n",
       "      <td>0.137987</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   HOT1A3       CIP     DSS-3   nitzsch   HOT5_F3      HP15  \\\n",
       "strain_nickname                                                               \n",
       "HOT1A3           0.000000  0.354890  2.472243  2.537937  2.475881  0.975858   \n",
       "CIP              0.354890  0.000000  2.486333  2.552027  2.489971  0.989948   \n",
       "DSS-3            2.472243  2.486333  0.000000  0.164756  0.106181  2.390807   \n",
       "nitzsch          2.537937  2.552027  0.164756  0.000000  0.171875  2.456501   \n",
       "HOT5_F3          2.475881  2.489971  0.106181  0.171875  0.000000  2.394445   \n",
       "HP15             0.975858  0.989948  2.390807  2.456501  2.394445  0.000000   \n",
       "HOT5_C3          2.474515  2.488605  0.137060  0.202754  0.140698  2.393078   \n",
       "Pgall            2.473170  2.487260  0.099989  0.157894  0.107108  2.391734   \n",
       "\n",
       "                  HOT5_C3     Pgall  \n",
       "strain_nickname                      \n",
       "HOT1A3           2.474515  2.473170  \n",
       "CIP              2.488605  2.487260  \n",
       "DSS-3            0.137060  0.099989  \n",
       "nitzsch          0.202754  0.157894  \n",
       "HOT5_F3          0.140698  0.107108  \n",
       "HP15             2.393078  2.391734  \n",
       "HOT5_C3          0.000000  0.137987  \n",
       "Pgall            0.137987  0.000000  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phylo_matrix_df =  phylo_matrix_df[metadata_df.strain_nickname]\n",
    "phylo_matrix_df = phylo_matrix_df.reindex(index=metadata_df.strain_nickname)\n",
    "phylo_matrix_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "de21cfbe-c50e-407b-b8be-48a8e8bdd46e",
   "metadata": {},
   "outputs": [],
   "source": [
    "dist_matrix = phylo_matrix_df\n",
    "title = 'Tree based on phylo distance'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "99163cdf-4c7d-4465-bc98-7a555d6311bb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HOT1A3</th>\n",
       "      <th>CIP</th>\n",
       "      <th>DSS-3</th>\n",
       "      <th>nitzsch</th>\n",
       "      <th>HOT5_F3</th>\n",
       "      <th>HP15</th>\n",
       "      <th>HOT5_C3</th>\n",
       "      <th>Pgall</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>strain_nickname</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HOT1A3</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.354890</td>\n",
       "      <td>2.472243</td>\n",
       "      <td>2.537937</td>\n",
       "      <td>2.475881</td>\n",
       "      <td>0.975858</td>\n",
       "      <td>2.474515</td>\n",
       "      <td>2.473170</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CIP</th>\n",
       "      <td>0.354890</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.486333</td>\n",
       "      <td>2.552027</td>\n",
       "      <td>2.489971</td>\n",
       "      <td>0.989948</td>\n",
       "      <td>2.488605</td>\n",
       "      <td>2.487260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DSS-3</th>\n",
       "      <td>2.472243</td>\n",
       "      <td>2.486333</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.164756</td>\n",
       "      <td>0.106181</td>\n",
       "      <td>2.390807</td>\n",
       "      <td>0.137060</td>\n",
       "      <td>0.099989</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>nitzsch</th>\n",
       "      <td>2.537937</td>\n",
       "      <td>2.552027</td>\n",
       "      <td>0.164756</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.171875</td>\n",
       "      <td>2.456501</td>\n",
       "      <td>0.202754</td>\n",
       "      <td>0.157894</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HOT5_F3</th>\n",
       "      <td>2.475881</td>\n",
       "      <td>2.489971</td>\n",
       "      <td>0.106181</td>\n",
       "      <td>0.171875</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.394445</td>\n",
       "      <td>0.140698</td>\n",
       "      <td>0.107108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HP15</th>\n",
       "      <td>0.975858</td>\n",
       "      <td>0.989948</td>\n",
       "      <td>2.390807</td>\n",
       "      <td>2.456501</td>\n",
       "      <td>2.394445</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.393078</td>\n",
       "      <td>2.391734</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HOT5_C3</th>\n",
       "      <td>2.474515</td>\n",
       "      <td>2.488605</td>\n",
       "      <td>0.137060</td>\n",
       "      <td>0.202754</td>\n",
       "      <td>0.140698</td>\n",
       "      <td>2.393078</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.137987</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pgall</th>\n",
       "      <td>2.473170</td>\n",
       "      <td>2.487260</td>\n",
       "      <td>0.099989</td>\n",
       "      <td>0.157894</td>\n",
       "      <td>0.107108</td>\n",
       "      <td>2.391734</td>\n",
       "      <td>0.137987</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   HOT1A3       CIP     DSS-3   nitzsch   HOT5_F3      HP15  \\\n",
       "strain_nickname                                                               \n",
       "HOT1A3           0.000000  0.354890  2.472243  2.537937  2.475881  0.975858   \n",
       "CIP              0.354890  0.000000  2.486333  2.552027  2.489971  0.989948   \n",
       "DSS-3            2.472243  2.486333  0.000000  0.164756  0.106181  2.390807   \n",
       "nitzsch          2.537937  2.552027  0.164756  0.000000  0.171875  2.456501   \n",
       "HOT5_F3          2.475881  2.489971  0.106181  0.171875  0.000000  2.394445   \n",
       "HP15             0.975858  0.989948  2.390807  2.456501  2.394445  0.000000   \n",
       "HOT5_C3          2.474515  2.488605  0.137060  0.202754  0.140698  2.393078   \n",
       "Pgall            2.473170  2.487260  0.099989  0.157894  0.107108  2.391734   \n",
       "\n",
       "                  HOT5_C3     Pgall  \n",
       "strain_nickname                      \n",
       "HOT1A3           2.474515  2.473170  \n",
       "CIP              2.488605  2.487260  \n",
       "DSS-3            0.137060  0.099989  \n",
       "nitzsch          0.202754  0.157894  \n",
       "HOT5_F3          0.140698  0.107108  \n",
       "HP15             2.393078  2.391734  \n",
       "HOT5_C3          0.000000  0.137987  \n",
       "Pgall            0.137987  0.000000  "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phylo_matrix_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "6e1f3d23-f764-49e3-85b1-7e0364e706d0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0     HOT1A3\n",
       "1        CIP\n",
       "2      DSS-3\n",
       "3    nitzsch\n",
       "5    HOT5_F3\n",
       "6       HP15\n",
       "7    HOT5_C3\n",
       "8      Pgall\n",
       "Name: strain_nickname, dtype: object"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata_df.strain_nickname.astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "5e56bbf2-f4df-4afa-8f6a-c8e6ba29078a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['HOT1A3', 'CIP', 'DSS-3', 'nitzsch', 'HOT5_F3', 'HP15', 'HOT5_C3',\n",
       "       'Pgall'],\n",
       "      dtype='object', name='strain_nickname')"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dist_matrix.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "f3058ca9-a633-4d97-953a-9cdf6fb6bef3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/svg+xml": [
       "<?xml version=\"1.0\" encoding=\"utf-8\" standalone=\"no\"?>\n",
       "<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n",
       "  \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n",
       "<svg xmlns:xlink=\"http://www.w3.org/1999/xlink\" width=\"460.46375pt\" height=\"291.6pt\" viewBox=\"0 0 460.46375 291.6\" xmlns=\"http://www.w3.org/2000/svg\" version=\"1.1\">\n",
       " <metadata>\n",
       "  <rdf:RDF xmlns:dc=\"http://purl.org/dc/elements/1.1/\" xmlns:cc=\"http://creativecommons.org/ns#\" xmlns:rdf=\"http://www.w3.org/1999/02/22-rdf-syntax-ns#\">\n",
       "   <cc:Work>\n",
       "    <dc:type rdf:resource=\"http://purl.org/dc/dcmitype/StillImage\"/>\n",
       "    <dc:date>2024-11-20T13:51:38.627301</dc:date>\n",
       "    <dc:format>image/svg+xml</dc:format>\n",
       "    <dc:creator>\n",
       "     <cc:Agent>\n",
       "      <dc:title>Matplotlib v3.8.0, https://matplotlib.org/</dc:title>\n",
       "     </cc:Agent>\n",
       "    </dc:creator>\n",
       "   </cc:Work>\n",
       "  </rdf:RDF>\n",
       " </metadata>\n",
       " <defs>\n",
       "  <style type=\"text/css\">*{stroke-linejoin: round; stroke-linecap: butt}</style>\n",
       " </defs>\n",
       " <g id=\"figure_1\">\n",
       "  <g id=\"patch_1\">\n",
       "   <path d=\"M 0 291.6 \n",
       "L 460.46375 291.6 \n",
       "L 460.46375 0 \n",
       "L 0 0 \n",
       "z\n",
       "\" style=\"fill: #ffffff\"/>\n",
       "  </g>\n",
       "  <g id=\"axes_1\">\n",
       "   <g id=\"patch_2\">\n",
       "    <path d=\"M 7.2 284.4 \n",
       "L 90.9 284.4 \n",
       "L 90.9 7.2 \n",
       "L 7.2 7.2 \n",
       "z\n",
       "\" style=\"fill: #ffffff\"/>\n",
       "   </g>\n",
       "   <g id=\"matplotlib.axis_1\"/>\n",
       "   <g id=\"matplotlib.axis_2\">\n",
       "    <g id=\"ytick_1\">\n",
       "     <g id=\"text_1\">\n",
       "      <text style=\"fill: #cc6677; font: 24px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"106.4\" y=\"275.810625\" transform=\"rotate(-0 106.4 275.810625)\">Sulfitobacter pseudonitzschiae</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_2\">\n",
       "     <g id=\"text_2\">\n",
       "      <text style=\"fill: #44aa99; font: 24px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"106.4\" y=\"241.168125\" transform=\"rotate(-0 106.4 241.168125)\">Roseovarius HOT5_C3</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_3\">\n",
       "     <g id=\"text_3\">\n",
       "      <text style=\"fill: #44aa99; font: 24px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"106.4\" y=\"206.518125\" transform=\"rotate(-0 106.4 206.518125)\">Marinovum HOT5_F3</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_4\">\n",
       "     <g id=\"text_4\">\n",
       "      <text style=\"fill: #cc6677; font: 24px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"106.4\" y=\"171.714375\" transform=\"rotate(-0 106.4 171.714375)\">Ruegeria pomeroyi</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_5\">\n",
       "     <g id=\"text_5\">\n",
       "      <text style=\"fill: #332288; font: 24px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"106.4\" y=\"137.064375\" transform=\"rotate(-0 106.4 137.064375)\">Phaeobacter gallaeciensis</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_6\">\n",
       "     <g id=\"text_6\">\n",
       "      <text style=\"fill: #332288; font: 24px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"106.4\" y=\"102.414375\" transform=\"rotate(-0 106.4 102.414375)\">Marinobacter adhaerens</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_7\">\n",
       "     <g id=\"text_7\">\n",
       "      <text style=\"fill: #882255; font: 24px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"106.4\" y=\"67.764375\" transform=\"rotate(-0 106.4 67.764375)\">Alteromonas macleodii</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_8\">\n",
       "     <g id=\"text_8\">\n",
       "      <text style=\"fill: #882255; font: 24px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"106.4\" y=\"33.114375\" transform=\"rotate(-0 106.4 33.114375)\">Pseudoalteromonas haloplanktis</text>\n",
       "     </g>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"LineCollection_1\">\n",
       "    <path d=\"M 90.9 163.125 \n",
       "L 87.566162 163.125 \n",
       "L 87.566162 128.475 \n",
       "L 90.9 128.475 \n",
       "\" clip-path=\"url(#p2a47b6b430)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "    <path d=\"M 90.9 197.775 \n",
       "L 87.359706 197.775 \n",
       "L 87.359706 145.8 \n",
       "L 87.566162 145.8 \n",
       "\" clip-path=\"url(#p2a47b6b430)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "    <path d=\"M 90.9 232.425 \n",
       "L 86.330145 232.425 \n",
       "L 86.330145 171.7875 \n",
       "L 87.359706 171.7875 \n",
       "\" clip-path=\"url(#p2a47b6b430)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "    <path d=\"M 90.9 267.075 \n",
       "L 85.635491 267.075 \n",
       "L 85.635491 202.10625 \n",
       "L 86.330145 202.10625 \n",
       "\" clip-path=\"url(#p2a47b6b430)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "    <path d=\"M 90.9 59.175 \n",
       "L 79.06725 59.175 \n",
       "L 79.06725 24.525 \n",
       "L 90.9 24.525 \n",
       "\" clip-path=\"url(#p2a47b6b430)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "    <path d=\"M 90.9 93.825 \n",
       "L 58.362937 93.825 \n",
       "L 58.362937 41.85 \n",
       "L 79.06725 41.85 \n",
       "\" clip-path=\"url(#p2a47b6b430)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "    <path d=\"M 85.635491 234.590625 \n",
       "L 11.185714 234.590625 \n",
       "L 11.185714 67.8375 \n",
       "L 58.362937 67.8375 \n",
       "\" clip-path=\"url(#p2a47b6b430)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "   </g>\n",
       "  </g>\n",
       " </g>\n",
       " <defs>\n",
       "  <clipPath id=\"p2a47b6b430\">\n",
       "   <rect x=\"7.2\" y=\"7.2\" width=\"83.7\" height=\"277.2\"/>\n",
       "  </clipPath>\n",
       " </defs>\n",
       "</svg>\n"
      ],
      "text/plain": [
       "<Figure size 150x500 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from matplotlib.pyplot import figure\n",
    "\n",
    "from scipy.cluster.hierarchy import dendrogram, linkage\n",
    "from scipy.spatial.distance import squareform\n",
    "\n",
    "\n",
    "figure(figsize=(1.5, 5))\n",
    "\n",
    "name_map = metadata_df.Name\n",
    "name_map.index= metadata_df.strain_nickname.astype(str)\n",
    "\n",
    "label_colors = metadata_df.Group.map(dict(zip(gorder,gpalette)))\n",
    "label_colors.index= metadata_df.Name\n",
    "\n",
    "dists = squareform(dist_matrix)\n",
    "linkage_matrix = linkage(dists, \"single\")\n",
    "dendrogram(\n",
    "    linkage_matrix, \n",
    "    labels=dist_matrix.index.map(name_map), \n",
    "    orientation='left', \n",
    "    link_color_func=lambda k: 'black',\n",
    "    leaf_font_size=24,\n",
    ")\n",
    "#plt.title(\"test\")\n",
    "\n",
    "\n",
    "ax = plt.gca()\n",
    "xlbls = ax.get_ymajorticklabels()\n",
    "for lbl in xlbls:\n",
    "    lbl.set_color(label_colors[lbl.get_text()])\n",
    "\n",
    "\n",
    "sns.despine(left=True, bottom=True)\n",
    "plt.xticks([])\n",
    "#plt.title(title)\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "7b1ccbfa-2e55-4483-b29b-a138c0da4d13",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/svg+xml": [
       "<?xml version=\"1.0\" encoding=\"utf-8\" standalone=\"no\"?>\n",
       "<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n",
       "  \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n",
       "<svg xmlns:xlink=\"http://www.w3.org/1999/xlink\" width=\"443.72375pt\" height=\"1012.32pt\" viewBox=\"0 0 443.72375 1012.32\" xmlns=\"http://www.w3.org/2000/svg\" version=\"1.1\">\n",
       " <metadata>\n",
       "  <rdf:RDF xmlns:dc=\"http://purl.org/dc/elements/1.1/\" xmlns:cc=\"http://creativecommons.org/ns#\" xmlns:rdf=\"http://www.w3.org/1999/02/22-rdf-syntax-ns#\">\n",
       "   <cc:Work>\n",
       "    <dc:type rdf:resource=\"http://purl.org/dc/dcmitype/StillImage\"/>\n",
       "    <dc:date>2024-11-20T13:51:38.783179</dc:date>\n",
       "    <dc:format>image/svg+xml</dc:format>\n",
       "    <dc:creator>\n",
       "     <cc:Agent>\n",
       "      <dc:title>Matplotlib v3.8.0, https://matplotlib.org/</dc:title>\n",
       "     </cc:Agent>\n",
       "    </dc:creator>\n",
       "   </cc:Work>\n",
       "  </rdf:RDF>\n",
       " </metadata>\n",
       " <defs>\n",
       "  <style type=\"text/css\">*{stroke-linejoin: round; stroke-linecap: butt}</style>\n",
       " </defs>\n",
       " <g id=\"figure_1\">\n",
       "  <g id=\"patch_1\">\n",
       "   <path d=\"M 0 1012.32 \n",
       "L 443.72375 1012.32 \n",
       "L 443.72375 -0 \n",
       "L 0 -0 \n",
       "z\n",
       "\" style=\"fill: #ffffff\"/>\n",
       "  </g>\n",
       "  <g id=\"axes_1\">\n",
       "   <g id=\"patch_2\">\n",
       "    <path d=\"M 7.2 1005.12 \n",
       "L 74.16 1005.12 \n",
       "L 74.16 7.2 \n",
       "L 7.2 7.2 \n",
       "z\n",
       "\" style=\"fill: #ffffff\"/>\n",
       "   </g>\n",
       "   <g id=\"matplotlib.axis_1\"/>\n",
       "   <g id=\"matplotlib.axis_2\">\n",
       "    <g id=\"ytick_1\">\n",
       "     <g id=\"text_1\">\n",
       "      <text style=\"fill: #cc6677; font: 24px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"89.66\" y=\"951.485625\" transform=\"rotate(-0 89.66 951.485625)\">Sulfitobacter pseudonitzschiae</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_2\">\n",
       "     <g id=\"text_2\">\n",
       "      <text style=\"fill: #44aa99; font: 24px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"89.66\" y=\"826.753125\" transform=\"rotate(-0 89.66 826.753125)\">Roseovarius HOT5_C3</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_3\">\n",
       "     <g id=\"text_3\">\n",
       "      <text style=\"fill: #44aa99; font: 24px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"89.66\" y=\"702.013125\" transform=\"rotate(-0 89.66 702.013125)\">Marinovum HOT5_F3</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_4\">\n",
       "     <g id=\"text_4\">\n",
       "      <text style=\"fill: #cc6677; font: 24px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"89.66\" y=\"577.119375\" transform=\"rotate(-0 89.66 577.119375)\">Ruegeria pomeroyi</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_5\">\n",
       "     <g id=\"text_5\">\n",
       "      <text style=\"fill: #332288; font: 24px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"89.66\" y=\"452.379375\" transform=\"rotate(-0 89.66 452.379375)\">Phaeobacter gallaeciensis</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_6\">\n",
       "     <g id=\"text_6\">\n",
       "      <text style=\"fill: #332288; font: 24px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"89.66\" y=\"327.639375\" transform=\"rotate(-0 89.66 327.639375)\">Marinobacter adhaerens</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_7\">\n",
       "     <g id=\"text_7\">\n",
       "      <text style=\"fill: #882255; font: 24px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"89.66\" y=\"202.899375\" transform=\"rotate(-0 89.66 202.899375)\">Alteromonas macleodii</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_8\">\n",
       "     <g id=\"text_8\">\n",
       "      <text style=\"fill: #882255; font: 24px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"89.66\" y=\"78.159375\" transform=\"rotate(-0 89.66 78.159375)\">Pseudoalteromonas haloplanktis</text>\n",
       "     </g>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"LineCollection_1\">\n",
       "    <path d=\"M 74.16 568.53 \n",
       "L 71.49293 568.53 \n",
       "L 71.49293 443.79 \n",
       "L 74.16 443.79 \n",
       "\" clip-path=\"url(#p41e86a90ce)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "    <path d=\"M 74.16 693.27 \n",
       "L 71.327764 693.27 \n",
       "L 71.327764 506.16 \n",
       "L 71.49293 506.16 \n",
       "\" clip-path=\"url(#p41e86a90ce)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "    <path d=\"M 74.16 818.01 \n",
       "L 70.504116 818.01 \n",
       "L 70.504116 599.715 \n",
       "L 71.327764 599.715 \n",
       "\" clip-path=\"url(#p41e86a90ce)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "    <path d=\"M 74.16 942.75 \n",
       "L 69.948393 942.75 \n",
       "L 69.948393 708.8625 \n",
       "L 70.504116 708.8625 \n",
       "\" clip-path=\"url(#p41e86a90ce)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "    <path d=\"M 74.16 194.31 \n",
       "L 64.6938 194.31 \n",
       "L 64.6938 69.57 \n",
       "L 74.16 69.57 \n",
       "\" clip-path=\"url(#p41e86a90ce)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "    <path d=\"M 74.16 319.05 \n",
       "L 48.13035 319.05 \n",
       "L 48.13035 131.94 \n",
       "L 64.6938 131.94 \n",
       "\" clip-path=\"url(#p41e86a90ce)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "    <path d=\"M 69.948393 825.80625 \n",
       "L 10.388571 825.80625 \n",
       "L 10.388571 225.495 \n",
       "L 48.13035 225.495 \n",
       "\" clip-path=\"url(#p41e86a90ce)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "   </g>\n",
       "  </g>\n",
       " </g>\n",
       " <defs>\n",
       "  <clipPath id=\"p41e86a90ce\">\n",
       "   <rect x=\"7.2\" y=\"7.2\" width=\"66.96\" height=\"997.92\"/>\n",
       "  </clipPath>\n",
       " </defs>\n",
       "</svg>\n"
      ],
      "text/plain": [
       "<Figure size 120x1800 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from matplotlib.pyplot import figure\n",
    "\n",
    "from scipy.cluster.hierarchy import dendrogram, linkage\n",
    "from scipy.spatial.distance import squareform\n",
    "\n",
    "\n",
    "figure(figsize=(1.2, 18))\n",
    "\n",
    "name_map = metadata_df.Name\n",
    "name_map.index= metadata_df.strain_nickname.astype(str)\n",
    "\n",
    "label_colors = metadata_df.Group.map(dict(zip(gorder,gpalette)))\n",
    "label_colors.index= metadata_df.Name\n",
    "\n",
    "dists = squareform(dist_matrix)\n",
    "linkage_matrix = linkage(dists, \"single\")\n",
    "dendrogram(\n",
    "    linkage_matrix, \n",
    "    labels=dist_matrix.index.map(name_map), \n",
    "    orientation='left', \n",
    "    link_color_func=lambda k: 'black',\n",
    "    leaf_font_size=24,\n",
    "    #facecolor='none',\n",
    ")\n",
    "#plt.title(\"test\")\n",
    "\n",
    "\n",
    "ax = plt.gca()\n",
    "xlbls = ax.get_ymajorticklabels()\n",
    "for lbl in xlbls:\n",
    "    lbl.set_color(label_colors[lbl.get_text()])\n",
    "\n",
    "\n",
    "sns.despine(left=True, bottom=True)\n",
    "plt.xticks([])\n",
    "#plt.title(title)\n",
    "save_fpath = os.path.join(save_dpath, 'fig_1_heterotroph_tree.svg')\n",
    "plt.savefig(save_fpath, format='svg', dpi=1200)\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "5c162daa-f88e-4d2b-9ef1-60cbf4d3c87e",
   "metadata": {},
   "outputs": [],
   "source": [
    "label_colors = metadata_df.Group.map(dict(zip(gorder,gpalette)))\n",
    "label_colors.index= metadata_df.Name\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "35e369a0-f002-42d5-b8d9-7fec02289d56",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Pseudoalteromonas haloplanktis',\n",
       " 'Alteromonas macleodii',\n",
       " 'Marinobacter adhaerens',\n",
       " 'Phaeobacter gallaeciensis',\n",
       " 'Ruegeria pomeroyi',\n",
       " 'Marinovum HOT5_F3',\n",
       " 'Roseovarius HOT5_C3',\n",
       " 'Sulfitobacter pseudonitzschiae']"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# tree order of strains\n",
    "list(reversed([lbl.get_text() for lbl in xlbls]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "72a896d1-6b8f-4850-93b2-b9189fff0401",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Strong',\n",
       " 'Strong',\n",
       " 'Inhibited',\n",
       " 'Inhibited',\n",
       " 'Sustained',\n",
       " 'Weak',\n",
       " 'Weak',\n",
       " 'Sustained']"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# tree order of groups\n",
    "\n",
    "groupmap = metadata_df.Group\n",
    "groupmap.index= metadata_df.Name\n",
    "\n",
    "list(reversed([groupmap[lbl.get_text()] for lbl in xlbls]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "8202abcd-e6d3-4560-b8a5-9b5fa705489b",
   "metadata": {},
   "outputs": [],
   "source": [
    "dist_matrix = gene_dist_matrix\n",
    "title = 'Tree based on genes distance'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "b277258c-36f4-481d-b105-c121d8a56c45",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/svg+xml": [
       "<?xml version=\"1.0\" encoding=\"utf-8\" standalone=\"no\"?>\n",
       "<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n",
       "  \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n",
       "<svg xmlns:xlink=\"http://www.w3.org/1999/xlink\" width=\"980.28pt\" height=\"247.41pt\" viewBox=\"0 0 980.28 247.41\" xmlns=\"http://www.w3.org/2000/svg\" version=\"1.1\">\n",
       " <metadata>\n",
       "  <rdf:RDF xmlns:dc=\"http://purl.org/dc/elements/1.1/\" xmlns:cc=\"http://creativecommons.org/ns#\" xmlns:rdf=\"http://www.w3.org/1999/02/22-rdf-syntax-ns#\">\n",
       "   <cc:Work>\n",
       "    <dc:type rdf:resource=\"http://purl.org/dc/dcmitype/StillImage\"/>\n",
       "    <dc:date>2024-11-20T13:51:38.944542</dc:date>\n",
       "    <dc:format>image/svg+xml</dc:format>\n",
       "    <dc:creator>\n",
       "     <cc:Agent>\n",
       "      <dc:title>Matplotlib v3.8.0, https://matplotlib.org/</dc:title>\n",
       "     </cc:Agent>\n",
       "    </dc:creator>\n",
       "   </cc:Work>\n",
       "  </rdf:RDF>\n",
       " </metadata>\n",
       " <defs>\n",
       "  <style type=\"text/css\">*{stroke-linejoin: round; stroke-linecap: butt}</style>\n",
       " </defs>\n",
       " <g id=\"figure_1\">\n",
       "  <g id=\"patch_1\">\n",
       "   <path d=\"M 0 247.41 \n",
       "L 980.28 247.41 \n",
       "L 980.28 -0 \n",
       "L 0 -0 \n",
       "z\n",
       "\" style=\"fill: #ffffff\"/>\n",
       "  </g>\n",
       "  <g id=\"axes_1\">\n",
       "   <g id=\"patch_2\">\n",
       "    <path d=\"M 406.44 240.21 \n",
       "L 573.84 240.21 \n",
       "L 573.84 73.89 \n",
       "L 406.44 73.89 \n",
       "z\n",
       "\" style=\"fill: #ffffff\"/>\n",
       "   </g>\n",
       "   <g id=\"matplotlib.axis_1\"/>\n",
       "   <g id=\"matplotlib.axis_2\">\n",
       "    <g id=\"ytick_1\">\n",
       "     <g id=\"text_1\">\n",
       "      <text style=\"fill: #44aa99; font: 12px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"589.34\" y=\"234.186562\" transform=\"rotate(-0 589.34 234.186562)\">Marinovum HOT5_F3</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_2\">\n",
       "     <g id=\"text_2\">\n",
       "      <text style=\"fill: #44aa99; font: 12px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"589.34\" y=\"213.396562\" transform=\"rotate(-0 589.34 213.396562)\">Roseovarius HOT5_C3</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_3\">\n",
       "     <g id=\"text_3\">\n",
       "      <text style=\"fill: #cc6677; font: 12px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"589.34\" y=\"192.602812\" transform=\"rotate(-0 589.34 192.602812)\">Sulfitobacter pseudonitzschiae</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_4\">\n",
       "     <g id=\"text_4\">\n",
       "      <text style=\"fill: #cc6677; font: 12px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"589.34\" y=\"171.739688\" transform=\"rotate(-0 589.34 171.739688)\">Ruegeria pomeroyi</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_5\">\n",
       "     <g id=\"text_5\">\n",
       "      <text style=\"fill: #332288; font: 12px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"589.34\" y=\"150.949687\" transform=\"rotate(-0 589.34 150.949687)\">Phaeobacter gallaeciensis</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_6\">\n",
       "     <g id=\"text_6\">\n",
       "      <text style=\"fill: #332288; font: 12px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"589.34\" y=\"130.159687\" transform=\"rotate(-0 589.34 130.159687)\">Marinobacter adhaerens</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_7\">\n",
       "     <g id=\"text_7\">\n",
       "      <text style=\"fill: #882255; font: 12px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"589.34\" y=\"109.369687\" transform=\"rotate(-0 589.34 109.369687)\">Alteromonas macleodii</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_8\">\n",
       "     <g id=\"text_8\">\n",
       "      <text style=\"fill: #882255; font: 12px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"589.34\" y=\"88.579688\" transform=\"rotate(-0 589.34 88.579688)\">Pseudoalteromonas haloplanktis</text>\n",
       "     </g>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"LineCollection_1\">\n",
       "    <path d=\"M 573.84 167.445 \n",
       "L 477.524337 167.445 \n",
       "L 477.524337 146.655 \n",
       "L 573.84 146.655 \n",
       "\" clip-path=\"url(#p7e76571003)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "    <path d=\"M 573.84 188.235 \n",
       "L 468.703442 188.235 \n",
       "L 468.703442 157.05 \n",
       "L 477.524337 157.05 \n",
       "\" clip-path=\"url(#p7e76571003)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "    <path d=\"M 573.84 209.025 \n",
       "L 462.040469 209.025 \n",
       "L 462.040469 172.6425 \n",
       "L 468.703442 172.6425 \n",
       "\" clip-path=\"url(#p7e76571003)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "    <path d=\"M 573.84 229.815 \n",
       "L 461.273181 229.815 \n",
       "L 461.273181 190.83375 \n",
       "L 462.040469 190.83375 \n",
       "\" clip-path=\"url(#p7e76571003)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "    <path d=\"M 573.84 105.075 \n",
       "L 464.194695 105.075 \n",
       "L 464.194695 84.285 \n",
       "L 573.84 84.285 \n",
       "\" clip-path=\"url(#p7e76571003)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "    <path d=\"M 573.84 125.865 \n",
       "L 435.254071 125.865 \n",
       "L 435.254071 94.68 \n",
       "L 464.194695 94.68 \n",
       "\" clip-path=\"url(#p7e76571003)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "    <path d=\"M 461.273181 210.324375 \n",
       "L 414.411429 210.324375 \n",
       "L 414.411429 110.2725 \n",
       "L 435.254071 110.2725 \n",
       "\" clip-path=\"url(#p7e76571003)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "   </g>\n",
       "   <g id=\"text_9\">\n",
       "    <text style=\"fill: #373737; font: 72px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: middle\" x=\"490.14\" y=\"58.73625\" transform=\"rotate(-0 490.14 58.73625)\">Tree based on genes distance</text>\n",
       "   </g>\n",
       "  </g>\n",
       " </g>\n",
       " <defs>\n",
       "  <clipPath id=\"p7e76571003\">\n",
       "   <rect x=\"406.44\" y=\"73.89\" width=\"167.4\" height=\"166.32\"/>\n",
       "  </clipPath>\n",
       " </defs>\n",
       "</svg>\n"
      ],
      "text/plain": [
       "<Figure size 300x300 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from matplotlib.pyplot import figure\n",
    "\n",
    "from scipy.cluster.hierarchy import dendrogram, linkage\n",
    "from scipy.spatial.distance import squareform\n",
    "\n",
    "\n",
    "figure(figsize=(3, 3))\n",
    "\n",
    "name_map = metadata_df.Name\n",
    "name_map.index= metadata_df.strain_nickname\n",
    "\n",
    "label_colors = metadata_df.Group.map(dict(zip(gorder,gpalette)))\n",
    "label_colors.index= metadata_df.Name\n",
    "\n",
    "dists = squareform(dist_matrix)\n",
    "linkage_matrix = linkage(dists, \"single\")\n",
    "dendrogram(\n",
    "    linkage_matrix, \n",
    "    labels=dist_matrix.index.map(name_map), \n",
    "    orientation='left', \n",
    "    link_color_func=lambda k: 'black'\n",
    ")\n",
    "#plt.title(\"test\")\n",
    "\n",
    "\n",
    "ax = plt.gca()\n",
    "xlbls = ax.get_ymajorticklabels()\n",
    "for lbl in xlbls:\n",
    "    lbl.set_color(label_colors[lbl.get_text()])\n",
    "\n",
    "\n",
    "sns.despine(left=True, bottom=True)\n",
    "plt.xticks([])\n",
    "plt.title(title)\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "cb77eec9-d84d-4bb2-8e1a-6ab654a95b62",
   "metadata": {},
   "outputs": [],
   "source": [
    "dist_matrix = trait_dist_matrix\n",
    "title = 'Tree based on trait distance'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "95fdcf51-b768-4224-85dd-cfca3aa9a043",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "63d97a4d-10ba-43ff-bb7a-65ef22937606",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/svg+xml": [
       "<?xml version=\"1.0\" encoding=\"utf-8\" standalone=\"no\"?>\n",
       "<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n",
       "  \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n",
       "<svg xmlns:xlink=\"http://www.w3.org/1999/xlink\" width=\"904.14pt\" height=\"246.56625pt\" viewBox=\"0 0 904.14 246.56625\" xmlns=\"http://www.w3.org/2000/svg\" version=\"1.1\">\n",
       " <metadata>\n",
       "  <rdf:RDF xmlns:dc=\"http://purl.org/dc/elements/1.1/\" xmlns:cc=\"http://creativecommons.org/ns#\" xmlns:rdf=\"http://www.w3.org/1999/02/22-rdf-syntax-ns#\">\n",
       "   <cc:Work>\n",
       "    <dc:type rdf:resource=\"http://purl.org/dc/dcmitype/StillImage\"/>\n",
       "    <dc:date>2024-11-20T13:51:39.057614</dc:date>\n",
       "    <dc:format>image/svg+xml</dc:format>\n",
       "    <dc:creator>\n",
       "     <cc:Agent>\n",
       "      <dc:title>Matplotlib v3.8.0, https://matplotlib.org/</dc:title>\n",
       "     </cc:Agent>\n",
       "    </dc:creator>\n",
       "   </cc:Work>\n",
       "  </rdf:RDF>\n",
       " </metadata>\n",
       " <defs>\n",
       "  <style type=\"text/css\">*{stroke-linejoin: round; stroke-linecap: butt}</style>\n",
       " </defs>\n",
       " <g id=\"figure_1\">\n",
       "  <g id=\"patch_1\">\n",
       "   <path d=\"M 0 246.56625 \n",
       "L 904.14 246.56625 \n",
       "L 904.14 0 \n",
       "L 0 0 \n",
       "z\n",
       "\" style=\"fill: #ffffff\"/>\n",
       "  </g>\n",
       "  <g id=\"axes_1\">\n",
       "   <g id=\"patch_2\">\n",
       "    <path d=\"M 368.37 239.36625 \n",
       "L 535.77 239.36625 \n",
       "L 535.77 73.04625 \n",
       "L 368.37 73.04625 \n",
       "z\n",
       "\" style=\"fill: #ffffff\"/>\n",
       "   </g>\n",
       "   <g id=\"matplotlib.axis_1\"/>\n",
       "   <g id=\"matplotlib.axis_2\">\n",
       "    <g id=\"ytick_1\">\n",
       "     <g id=\"text_1\">\n",
       "      <text style=\"fill: #44aa99; font: 12px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"551.27\" y=\"233.342813\" transform=\"rotate(-0 551.27 233.342813)\">Marinovum HOT5_F3</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_2\">\n",
       "     <g id=\"text_2\">\n",
       "      <text style=\"fill: #cc6677; font: 12px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"551.27\" y=\"212.475938\" transform=\"rotate(-0 551.27 212.475938)\">Ruegeria pomeroyi</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_3\">\n",
       "     <g id=\"text_3\">\n",
       "      <text style=\"fill: #44aa99; font: 12px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"551.27\" y=\"191.762813\" transform=\"rotate(-0 551.27 191.762813)\">Roseovarius HOT5_C3</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_4\">\n",
       "     <g id=\"text_4\">\n",
       "      <text style=\"fill: #cc6677; font: 12px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"551.27\" y=\"170.969063\" transform=\"rotate(-0 551.27 170.969063)\">Sulfitobacter pseudonitzschiae</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_5\">\n",
       "     <g id=\"text_5\">\n",
       "      <text style=\"fill: #332288; font: 12px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"551.27\" y=\"150.105938\" transform=\"rotate(-0 551.27 150.105938)\">Phaeobacter gallaeciensis</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_6\">\n",
       "     <g id=\"text_6\">\n",
       "      <text style=\"fill: #332288; font: 12px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"551.27\" y=\"129.315938\" transform=\"rotate(-0 551.27 129.315938)\">Marinobacter adhaerens</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_7\">\n",
       "     <g id=\"text_7\">\n",
       "      <text style=\"fill: #882255; font: 12px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"551.27\" y=\"108.525937\" transform=\"rotate(-0 551.27 108.525937)\">Alteromonas macleodii</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_8\">\n",
       "     <g id=\"text_8\">\n",
       "      <text style=\"fill: #882255; font: 12px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"551.27\" y=\"87.735938\" transform=\"rotate(-0 551.27 87.735938)\">Pseudoalteromonas haloplanktis</text>\n",
       "     </g>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"LineCollection_1\">\n",
       "    <path d=\"M 535.77 208.18125 \n",
       "L 452.342608 208.18125 \n",
       "L 452.342608 187.39125 \n",
       "L 535.77 187.39125 \n",
       "\" clip-path=\"url(#pd9aa4f9731)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "    <path d=\"M 535.77 166.60125 \n",
       "L 450.96851 166.60125 \n",
       "L 450.96851 145.81125 \n",
       "L 535.77 145.81125 \n",
       "\" clip-path=\"url(#pd9aa4f9731)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "    <path d=\"M 452.342608 197.78625 \n",
       "L 449.678105 197.78625 \n",
       "L 449.678105 156.20625 \n",
       "L 450.96851 156.20625 \n",
       "\" clip-path=\"url(#pd9aa4f9731)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "    <path d=\"M 535.77 228.97125 \n",
       "L 436.29712 228.97125 \n",
       "L 436.29712 176.99625 \n",
       "L 449.678105 176.99625 \n",
       "\" clip-path=\"url(#pd9aa4f9731)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "    <path d=\"M 535.77 104.23125 \n",
       "L 446.515142 104.23125 \n",
       "L 446.515142 83.44125 \n",
       "L 535.77 83.44125 \n",
       "\" clip-path=\"url(#pd9aa4f9731)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "    <path d=\"M 535.77 125.02125 \n",
       "L 397.315051 125.02125 \n",
       "L 397.315051 93.83625 \n",
       "L 446.515142 93.83625 \n",
       "\" clip-path=\"url(#pd9aa4f9731)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "    <path d=\"M 436.29712 202.98375 \n",
       "L 376.341429 202.98375 \n",
       "L 376.341429 109.42875 \n",
       "L 397.315051 109.42875 \n",
       "\" clip-path=\"url(#pd9aa4f9731)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "   </g>\n",
       "   <g id=\"text_9\">\n",
       "    <text style=\"fill: #373737; font: 72px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: middle\" x=\"452.07\" y=\"58.73625\" transform=\"rotate(-0 452.07 58.73625)\">Tree based on trait distance</text>\n",
       "   </g>\n",
       "  </g>\n",
       " </g>\n",
       " <defs>\n",
       "  <clipPath id=\"pd9aa4f9731\">\n",
       "   <rect x=\"368.37\" y=\"73.04625\" width=\"167.4\" height=\"166.32\"/>\n",
       "  </clipPath>\n",
       " </defs>\n",
       "</svg>\n"
      ],
      "text/plain": [
       "<Figure size 300x300 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from matplotlib.pyplot import figure\n",
    "\n",
    "from scipy.cluster.hierarchy import dendrogram, linkage\n",
    "from scipy.spatial.distance import squareform\n",
    "\n",
    "\n",
    "figure(figsize=(3, 3))\n",
    "\n",
    "name_map = metadata_df.Name\n",
    "name_map.index= metadata_df.strain_nickname\n",
    "\n",
    "label_colors = metadata_df.Group.map(dict(zip(gorder,gpalette)))\n",
    "label_colors.index= metadata_df.Name\n",
    "\n",
    "dists = squareform(dist_matrix)\n",
    "linkage_matrix = linkage(dists, \"single\")\n",
    "dendrogram(\n",
    "    linkage_matrix, \n",
    "    labels=dist_matrix.index.map(name_map), \n",
    "    orientation='left', \n",
    "    link_color_func=lambda k: 'black'\n",
    ")\n",
    "#plt.title(\"test\")\n",
    "\n",
    "\n",
    "ax = plt.gca()\n",
    "xlbls = ax.get_ymajorticklabels()\n",
    "for lbl in xlbls:\n",
    "    lbl.set_color(label_colors[lbl.get_text()])\n",
    "\n",
    "\n",
    "sns.despine(left=True, bottom=True)\n",
    "plt.xticks([])\n",
    "plt.title(title)\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "4586795b-9b37-4dc0-a99c-3095ffeb217d",
   "metadata": {},
   "outputs": [],
   "source": [
    "dist_matrix = phylo_dist_matrix\n",
    "title = 'Tree based on phylo distance'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "7bb24f7a-3de1-4376-8af2-44ca4d72275f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/svg+xml": [
       "<?xml version=\"1.0\" encoding=\"utf-8\" standalone=\"no\"?>\n",
       "<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n",
       "  \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n",
       "<svg xmlns:xlink=\"http://www.w3.org/1999/xlink\" width=\"956.23875pt\" height=\"247.41pt\" viewBox=\"0 0 956.23875 247.41\" xmlns=\"http://www.w3.org/2000/svg\" version=\"1.1\">\n",
       " <metadata>\n",
       "  <rdf:RDF xmlns:dc=\"http://purl.org/dc/elements/1.1/\" xmlns:cc=\"http://creativecommons.org/ns#\" xmlns:rdf=\"http://www.w3.org/1999/02/22-rdf-syntax-ns#\">\n",
       "   <cc:Work>\n",
       "    <dc:type rdf:resource=\"http://purl.org/dc/dcmitype/StillImage\"/>\n",
       "    <dc:date>2024-11-20T13:51:39.162055</dc:date>\n",
       "    <dc:format>image/svg+xml</dc:format>\n",
       "    <dc:creator>\n",
       "     <cc:Agent>\n",
       "      <dc:title>Matplotlib v3.8.0, https://matplotlib.org/</dc:title>\n",
       "     </cc:Agent>\n",
       "    </dc:creator>\n",
       "   </cc:Work>\n",
       "  </rdf:RDF>\n",
       " </metadata>\n",
       " <defs>\n",
       "  <style type=\"text/css\">*{stroke-linejoin: round; stroke-linecap: butt}</style>\n",
       " </defs>\n",
       " <g id=\"figure_1\">\n",
       "  <g id=\"patch_1\">\n",
       "   <path d=\"M 0 247.41 \n",
       "L 956.23875 247.41 \n",
       "L 956.23875 -0 \n",
       "L 0 -0 \n",
       "z\n",
       "\" style=\"fill: #ffffff\"/>\n",
       "  </g>\n",
       "  <g id=\"axes_1\">\n",
       "   <g id=\"patch_2\">\n",
       "    <path d=\"M 394.419375 240.21 \n",
       "L 561.819375 240.21 \n",
       "L 561.819375 73.89 \n",
       "L 394.419375 73.89 \n",
       "z\n",
       "\" style=\"fill: #ffffff\"/>\n",
       "   </g>\n",
       "   <g id=\"matplotlib.axis_1\"/>\n",
       "   <g id=\"matplotlib.axis_2\">\n",
       "    <g id=\"ytick_1\">\n",
       "     <g id=\"text_1\">\n",
       "      <text style=\"fill: #cc6677; font: 12px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"577.319375\" y=\"230.717812\" transform=\"rotate(-0 577.319375 230.717812)\">Sulfitobacter pseudonitzschiae</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_2\">\n",
       "     <g id=\"text_2\">\n",
       "      <text style=\"fill: #cc6677; font: 12px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"577.319375\" y=\"202.924688\" transform=\"rotate(-0 577.319375 202.924688)\">Ruegeria pomeroyi</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_3\">\n",
       "     <g id=\"text_3\">\n",
       "      <text style=\"fill: #332288; font: 12px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"577.319375\" y=\"175.204687\" transform=\"rotate(-0 577.319375 175.204687)\">Phaeobacter gallaeciensis</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_4\">\n",
       "     <g id=\"text_4\">\n",
       "      <text style=\"fill: #332288; font: 12px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"577.319375\" y=\"147.484688\" transform=\"rotate(-0 577.319375 147.484688)\">Marinobacter adhaerens</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_5\">\n",
       "     <g id=\"text_5\">\n",
       "      <text style=\"fill: #882255; font: 12px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"577.319375\" y=\"119.764687\" transform=\"rotate(-0 577.319375 119.764687)\">Pseudoalteromonas haloplanktis</text>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_6\">\n",
       "     <g id=\"text_6\">\n",
       "      <text style=\"fill: #882255; font: 12px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: start\" x=\"577.319375\" y=\"92.044688\" transform=\"rotate(-0 577.319375 92.044688)\">Alteromonas macleodii</text>\n",
       "     </g>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"LineCollection_1\">\n",
       "    <path d=\"M 561.819375 198.63 \n",
       "L 545.128064 198.63 \n",
       "L 545.128064 170.91 \n",
       "L 561.819375 170.91 \n",
       "\" clip-path=\"url(#p98068407b5)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "    <path d=\"M 561.819375 226.35 \n",
       "L 536.07394 226.35 \n",
       "L 536.07394 184.77 \n",
       "L 545.128064 184.77 \n",
       "\" clip-path=\"url(#p98068407b5)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "    <path d=\"M 561.819375 115.47 \n",
       "L 516.543778 115.47 \n",
       "L 516.543778 87.75 \n",
       "L 561.819375 87.75 \n",
       "\" clip-path=\"url(#p98068407b5)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "    <path d=\"M 561.819375 143.19 \n",
       "L 482.341522 143.19 \n",
       "L 482.341522 101.61 \n",
       "L 516.543778 101.61 \n",
       "\" clip-path=\"url(#p98068407b5)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "    <path d=\"M 536.07394 205.56 \n",
       "L 402.390804 205.56 \n",
       "L 402.390804 122.4 \n",
       "L 482.341522 122.4 \n",
       "\" clip-path=\"url(#p98068407b5)\" style=\"fill: none; stroke: #000000; stroke-width: 3\"/>\n",
       "   </g>\n",
       "   <g id=\"text_7\">\n",
       "    <text style=\"fill: #373737; font: 72px 'Arial', 'DejaVu Sans', 'Liberation Sans', 'Bitstream Vera Sans', sans-serif; text-anchor: middle\" x=\"478.119375\" y=\"58.73625\" transform=\"rotate(-0 478.119375 58.73625)\">Tree based on phylo distance</text>\n",
       "   </g>\n",
       "  </g>\n",
       " </g>\n",
       " <defs>\n",
       "  <clipPath id=\"p98068407b5\">\n",
       "   <rect x=\"394.419375\" y=\"73.89\" width=\"167.4\" height=\"166.32\"/>\n",
       "  </clipPath>\n",
       " </defs>\n",
       "</svg>\n"
      ],
      "text/plain": [
       "<Figure size 300x300 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from matplotlib.pyplot import figure\n",
    "\n",
    "from scipy.cluster.hierarchy import dendrogram, linkage\n",
    "from scipy.spatial.distance import squareform\n",
    "\n",
    "\n",
    "figure(figsize=(3, 3))\n",
    "\n",
    "name_map = metadata_df.Name\n",
    "name_map.index= metadata_df.strain_nickname\n",
    "\n",
    "label_colors = metadata_df.Group.map(dict(zip(gorder,gpalette)))\n",
    "label_colors.index= metadata_df.Name\n",
    "\n",
    "dists = squareform(dist_matrix)\n",
    "linkage_matrix = linkage(dists, \"single\")\n",
    "dendrogram(\n",
    "    linkage_matrix, \n",
    "    labels=dist_matrix.index.map(name_map), \n",
    "    orientation='left', \n",
    "    link_color_func=lambda k: 'black'\n",
    ")\n",
    "#plt.title(\"test\")\n",
    "\n",
    "\n",
    "ax = plt.gca()\n",
    "xlbls = ax.get_ymajorticklabels()\n",
    "for lbl in xlbls:\n",
    "    lbl.set_color(label_colors[lbl.get_text()])\n",
    "\n",
    "\n",
    "sns.despine(left=True, bottom=True)\n",
    "plt.xticks([])\n",
    "plt.title(title)\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1cc4a9b0-012a-4cd5-95f4-3e852988358a",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
