{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "0bae4279-0be6-4de9-bfbd-10b422b203f6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# genomic traits analysis for the strains used in the experiment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "c4d139fe-1d6e-46f6-9adf-3937ec1982b0",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import pprint\n",
    "sns.set(style=\"white\", context='poster')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "a24632d9-843b-414e-a36e-b7b997804ccb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['42003_2022_3184_MOESM4_ESM_select_genomes.xlsx',\n",
       " 'genetic_traits_table.tsv',\n",
       " 'genetic_traits_table_filtered.tsv',\n",
       " 'genomes_annotation_long_format.7z',\n",
       " 'strain_list.csv']"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "dpath = 'trait_gene_annotations'\n",
    "os.listdir(dpath)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "33953a01-7ca7-41b9-aaa9-2894808174cd",
   "metadata": {},
   "outputs": [],
   "source": [
    "strains = pd.read_csv(os.path.join(dpath, 'strain_list.csv'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "357dfa08-39d3-4cfd-834a-d01615453ef6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Species</th>\n",
       "      <th>GTDB taxonomy</th>\n",
       "      <th>NCBI taxonID</th>\n",
       "      <th>Accession number</th>\n",
       "      <th>Filename</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Ruegeria pomeroyi DSS-3</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>246200</td>\n",
       "      <td>GCA_000011965.2</td>\n",
       "      <td>GCA_000011965</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Pseudoalteromonas haloplanktis TAC125</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>326442</td>\n",
       "      <td>GCA_000026085.1</td>\n",
       "      <td>GCA_000026085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Alteromonas macleodii HOT1A3</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>529120</td>\n",
       "      <td>2687454166</td>\n",
       "      <td>2687453488</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Marinobacter adhaerens HP15</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>225937</td>\n",
       "      <td>650377991</td>\n",
       "      <td>650377991</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Phaeobacter gallaeciensis DSM 26640</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>1423144</td>\n",
       "      <td>GCA_000511385.1</td>\n",
       "      <td>GCA_000511385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Sulfitobacter pseudonitzschiae strain SMR1</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>1402135</td>\n",
       "      <td>CeMEB_private</td>\n",
       "      <td>Sulfitobacter_pseudonitzschiae_SMR1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      Species  \\\n",
       "0                     Ruegeria pomeroyi DSS-3   \n",
       "1       Pseudoalteromonas haloplanktis TAC125   \n",
       "2                Alteromonas macleodii HOT1A3   \n",
       "3                 Marinobacter adhaerens HP15   \n",
       "4         Phaeobacter gallaeciensis DSM 26640   \n",
       "5  Sulfitobacter pseudonitzschiae strain SMR1   \n",
       "\n",
       "                                       GTDB taxonomy  NCBI taxonID  \\\n",
       "0  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...        246200   \n",
       "1  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...        326442   \n",
       "2  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...        529120   \n",
       "3  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...        225937   \n",
       "4  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...       1423144   \n",
       "5  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...       1402135   \n",
       "\n",
       "  Accession number                             Filename  \n",
       "0  GCA_000011965.2                        GCA_000011965  \n",
       "1  GCA_000026085.1                        GCA_000026085  \n",
       "2       2687454166                           2687453488  \n",
       "3        650377991                            650377991  \n",
       "4  GCA_000511385.1                        GCA_000511385  \n",
       "5    CeMEB_private  Sulfitobacter_pseudonitzschiae_SMR1  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "strains"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "733907de-7db5-42af-b7a9-1e89a49bb32c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>GFC</th>\n",
       "      <th>Species</th>\n",
       "      <th>Taxon</th>\n",
       "      <th>Filename</th>\n",
       "      <th>B1-synt</th>\n",
       "      <th>B1-trans</th>\n",
       "      <th>B12-synt</th>\n",
       "      <th>B12-trans</th>\n",
       "      <th>B7-synt</th>\n",
       "      <th>B7-trans</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>gamma proteobacterium SCGC AAA076-D02</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>2228664052</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gamma proteobacterium SCGC AAA076-F14</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>2236347022</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAR324 cluster deltaproteobacterium REDSEA-S10_B5</td>\n",
       "      <td>Others</td>\n",
       "      <td>2651870254</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>alpha proteobacterium sp. HIMB59</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>2503982041</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>alpha proteobacterium SCGC AAA240-E13</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>2236661011</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0  GFC                                            Species  \\\n",
       "0         NaN  1.0              gamma proteobacterium SCGC AAA076-D02   \n",
       "1         NaN  NaN              gamma proteobacterium SCGC AAA076-F14   \n",
       "2         NaN  NaN  SAR324 cluster deltaproteobacterium REDSEA-S10_B5   \n",
       "3         NaN  2.0                   alpha proteobacterium sp. HIMB59   \n",
       "4         NaN  NaN              alpha proteobacterium SCGC AAA240-E13   \n",
       "\n",
       "                 Taxon    Filename  B1-synt  B1-trans  B12-synt  B12-trans  \\\n",
       "0  Gammaproteobacteria  2228664052        1         0         0          0   \n",
       "1  Gammaproteobacteria  2236347022        1         0         0          0   \n",
       "2               Others  2651870254        0         1         0          0   \n",
       "3  Alphaproteobacteria  2503982041        0         1         0          0   \n",
       "4  Alphaproteobacteria  2236661011        1         0         0          0   \n",
       "\n",
       "   B7-synt  B7-trans  \n",
       "0        1         0  \n",
       "1        1         0  \n",
       "2        0         1  \n",
       "3        0         1  \n",
       "4        0         1  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vitamins_df = pd.read_excel(os.path.join(dpath, '42003_2022_3184_MOESM4_ESM_select_genomes.xlsx'), sheet_name='Supp_Data_4', skiprows=3)\n",
    "vitamins_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "c4260a2c-919a-4a9e-8e0b-9f0e7da0959a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>ID</th>\n",
       "      <th>Description</th>\n",
       "      <th>Frequency</th>\n",
       "      <th>Interaction traits</th>\n",
       "      <th>KEGG hierarchy</th>\n",
       "      <th>...6</th>\n",
       "      <th>Notes</th>\n",
       "      <th>Reference</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>M00001</td>\n",
       "      <td>M00001  Glycolysis (Embden-Meyerhof pathway), ...</td>\n",
       "      <td>0.484144</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Carbohydrate metabolism</td>\n",
       "      <td>Central carbohydrate metabolism</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>M00002</td>\n",
       "      <td>M00002  Glycolysis, core module involving thre...</td>\n",
       "      <td>0.995772</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Carbohydrate metabolism</td>\n",
       "      <td>Central carbohydrate metabolism</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>M00003</td>\n",
       "      <td>M00003  Gluconeogenesis, oxaloacetate =&gt; fruct...</td>\n",
       "      <td>0.813953</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Carbohydrate metabolism</td>\n",
       "      <td>Central carbohydrate metabolism</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>M00004</td>\n",
       "      <td>M00004  Pentose phosphate pathway (Pentose pho...</td>\n",
       "      <td>0.773784</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Carbohydrate metabolism</td>\n",
       "      <td>Central carbohydrate metabolism</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>M00005</td>\n",
       "      <td>M00005  PRPP biosynthesis, ribose 5P =&gt; PRPP [...</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Carbohydrate metabolism</td>\n",
       "      <td>Central carbohydrate metabolism</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0      ID                                        Description  \\\n",
       "0         NaN  M00001  M00001  Glycolysis (Embden-Meyerhof pathway), ...   \n",
       "1         NaN  M00002  M00002  Glycolysis, core module involving thre...   \n",
       "2         NaN  M00003  M00003  Gluconeogenesis, oxaloacetate => fruct...   \n",
       "3         NaN  M00004  M00004  Pentose phosphate pathway (Pentose pho...   \n",
       "4         NaN  M00005  M00005  PRPP biosynthesis, ribose 5P => PRPP [...   \n",
       "\n",
       "   Frequency Interaction traits           KEGG hierarchy  \\\n",
       "0   0.484144                NaN  Carbohydrate metabolism   \n",
       "1   0.995772                NaN  Carbohydrate metabolism   \n",
       "2   0.813953                NaN  Carbohydrate metabolism   \n",
       "3   0.773784                NaN  Carbohydrate metabolism   \n",
       "4   1.000000                NaN  Carbohydrate metabolism   \n",
       "\n",
       "                              ...6 Notes Reference  \n",
       "0  Central carbohydrate metabolism   NaN       NaN  \n",
       "1  Central carbohydrate metabolism   NaN       NaN  \n",
       "2  Central carbohydrate metabolism   NaN       NaN  \n",
       "3  Central carbohydrate metabolism   NaN       NaN  \n",
       "4  Central carbohydrate metabolism   NaN       NaN  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kegg_names_df = pd.read_excel(os.path.join(dpath, '42003_2022_3184_MOESM4_ESM_select_genomes.xlsx'), sheet_name='Supp_Data_6', skiprows=3)\n",
    "kegg_names_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "d0f34870-eba3-42de-8b03-3cd403f87cc9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>filename</th>\n",
       "      <th>M00001</th>\n",
       "      <th>M00002</th>\n",
       "      <th>M00003</th>\n",
       "      <th>M00004</th>\n",
       "      <th>M00005</th>\n",
       "      <th>M00006</th>\n",
       "      <th>M00007</th>\n",
       "      <th>M00008</th>\n",
       "      <th>M00009</th>\n",
       "      <th>...</th>\n",
       "      <th>pvuABCDE</th>\n",
       "      <th>pvsABCDE</th>\n",
       "      <th>Auxin.TAM</th>\n",
       "      <th>Auxin.Ipyr</th>\n",
       "      <th>Auxin.IAN</th>\n",
       "      <th>Auxin.IAM</th>\n",
       "      <th>DMSP_demethylation</th>\n",
       "      <th>DMSP_cleavage</th>\n",
       "      <th>DHPS_catabolism</th>\n",
       "      <th>Taurine_in_TCA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2228664052.fna</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2236347022.fna</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2236661011.fna</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2501651206.fna</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2501651207.fna</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 579 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         filename  M00001  M00002  M00003  M00004  M00005  M00006  M00007  \\\n",
       "0  2228664052.fna       0       1       1       0       1       0       1   \n",
       "1  2236347022.fna       0       1       1       0       1       1       1   \n",
       "2  2236661011.fna       0       0       0       0       1       0       1   \n",
       "3  2501651206.fna       1       1       1       0       1       0       1   \n",
       "4  2501651207.fna       0       1       1       1       1       1       1   \n",
       "\n",
       "   M00008  M00009  ...  pvuABCDE  pvsABCDE  Auxin.TAM  Auxin.Ipyr  Auxin.IAN  \\\n",
       "0       0       0  ...         0         0          1           1          0   \n",
       "1       0       0  ...         0         0          0           1          0   \n",
       "2       0       1  ...         0         0          1           0          0   \n",
       "3       0       0  ...         1         0          1           1          0   \n",
       "4       1       1  ...         1         0          0           0          0   \n",
       "\n",
       "   Auxin.IAM  DMSP_demethylation  DMSP_cleavage  DHPS_catabolism  \\\n",
       "0          0                   1              0                1   \n",
       "1          0                   1              0                1   \n",
       "2          0                   1              0                1   \n",
       "3          0                   0              0                0   \n",
       "4          0                   0              0                0   \n",
       "\n",
       "   Taurine_in_TCA  \n",
       "0               0  \n",
       "1               1  \n",
       "2               0  \n",
       "3               0  \n",
       "4               0  \n",
       "\n",
       "[5 rows x 579 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "traits_df = pd.read_csv(os.path.join(dpath, 'genetic_traits_table.tsv'), sep='\\t')\n",
    "traits_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "9c3c80a0-ecc7-4e3d-aa02-8e3e45273713",
   "metadata": {},
   "outputs": [],
   "source": [
    "traits_df['filename'] = traits_df.filename.str.replace('.fna', '', regex=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cb8464dd-0f35-487f-831b-3870cb01ee85",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
